--- cminpack-1.1.3/CMakeLists.txt	2011-02-06 20:49:15.000000000 +0100
+++ cminpack-1.1.3.mine/CMakeLists.txt	2011-04-25 23:23:30.613386326 +0200
@@ -5,6 +5,7 @@
 project (CMINPACK)
 string(TOLOWER ${PROJECT_NAME} PROJECT_NAME_LOWER)
 
+set(CMAKE_INSTALL_PREFIX /usr)
 include(${PROJECT_SOURCE_DIR}/cmake/cminpack_utils.cmake)
 # Set version and OS-specific settings
 set(CMINPACK_VERSION 1.0.90 CACHE STRING "CMinpack version")
@@ -32,6 +33,9 @@
 else (SHARED_LIBS)
   message (STATUS "Building static libraries.")
   set (LIB_TYPE STATIC)
+	if(WIN32)
+		add_definitions(-DCMINPACK_NO_DLL)
+	endif(WIN32)
 endif (SHARED_LIBS)
 
 #set(CMAKE_INSTALL_PREFIX ${PROJECT_SOURCE_DIR}/../build)
@@ -52,9 +56,15 @@
 set (cminpack_hdrs
     cminpack.h minpack.h)
 
-add_library (cminpack ${LIB_TYPE} ${cminpack_srcs})
+add_library (cminpack SHARED ${cminpack_srcs})
+add_library (cminpack_static STATIC ${cminpack_srcs}) 
+set_target_properties(cminpack_static  PROPERTIES ARCHIVE_OUTPUT_NAME cminpack)
+add_library (cminpack-debug SHARED ${cminpack_srcs})
+set_target_properties(cminpack-debug PROPERTIES OUTPUT_NAME cminpack-gd COMPILE_FLAGS ${CMAKE_CXX_FLAGS_DEBUG})
+add_library (cminpack_static-debug STATIC ${cminpack_srcs}) 
+set_target_properties(cminpack_static-debug PROPERTIES OUTPUT_NAME cminpack-gd COMPILE_FLAGS ${CMAKE_CXX_FLAGS_DEBUG})
 
-install (TARGETS cminpack 
+install (TARGETS cminpack cminpack_static cminpack-debug cminpack_static-debug
    LIBRARY DESTINATION ${CMINPACK_LIB_INSTALL_DIR} COMPONENT library
    ARCHIVE DESTINATION ${CMINPACK_LIB_INSTALL_DIR} COMPONENT library
    RUNTIME DESTINATION ${CMINPACK_LIB_INSTALL_DIR} COMPONENT library)
@@ -67,3 +77,34 @@
 
 set_target_properties(cminpack PROPERTIES VERSION ${CMINPACK_VERSION})
 
+# RPM
+find_program(RPM_PROGRAM rpm)
+if(EXISTS ${RPM_PROGRAM})
+  list(APPEND CPACK_GENERATOR "RPM")
+endif(EXISTS ${RPM_PROGRAM})
+# DEB
+find_program(DPKG_PROGRAM dpkg)
+if(EXISTS ${DPKG_PROGRAM})
+  list(APPEND CPACK_GENERATOR "DEB")
+endif(EXISTS ${DPKG_PROGRAM})
+# NSIS
+find_program(NSIS_PROGRAM makensis MakeNSIS)
+if(EXISTS ${NSIS_PROGRAM})
+  list(APPEND CPACK_GENERATOR "NSIS")
+endif(EXISTS ${NSIS_PROGRAM})
+# dpkg
+find_program(PACKAGE_MAKER_PROGRAM PackageMaker
+	    HINTS /Developer/Applications/Utilities)
+if(EXISTS ${PACKAGE_MAKER_PROGRAM})
+  list(APPEND CPACK_GENERATOR "PackageMaker")
+endif(EXISTS ${PACKAGE_MAKER_PROGRAM})
+ 
+set(CPACK_GENERATOR "${CPACK_GENERATOR}")
+set(CPACK_MONOLITHIC_INSTALL 1)
+set(CPACK_PACKAGE_VERSION_MAJOR "@CMINPACK_MAJOR_VERSION@")
+set(CPACK_PACKAGE_VERSION_MINOR "@CMINPACK_MINOR_VERSION@")
+set(CPACK_PACKAGE_VERSION_PATCH "@CMINPACK_REVISION_VERSION@")
+include(InstallRequiredSystemLibraries)
+set(CPACK_PACKAGE_CONTACT "Frederic Devernay")
+include(CPack)
+
