--- cminpack-1.1.3/CMakeLists.txt	2011-04-15 07:28:11.841140175 +0000
+++ cminpack-1.1.3/CMakeLists.txt	2011-04-15 07:28:38.841934851 +0000
@@ -67,3 +67,35 @@
 
 set_target_properties(cminpack PROPERTIES VERSION ${CMINPACK_VERSION})
 
+# RPM
+find_program(RPM_PROGRAM rpm)
+if(EXISTS ${RPM_PROGRAM})
+  list(APPEND CPACK_GENERATOR "RPM")
+endif(EXISTS ${RPM_PROGRAM})
+# DEB
+find_program(DPKG_PROGRAM dpkg)
+if(EXISTS ${DPKG_PROGRAM})
+  list(APPEND CPACK_GENERATOR "DEB")
+endif(EXISTS ${DPKG_PROGRAM})
+# NSIS
+find_program(NSIS_PROGRAM makensis MakeNSIS)
+if(EXISTS ${NSIS_PROGRAM})
+  list(APPEND CPACK_GENERATOR "NSIS")
+endif(EXISTS ${NSIS_PROGRAM})
+# dpkg
+find_program(PACKAGE_MAKER_PROGRAM PackageMaker
+	    HINTS /Developer/Applications/Utilities)
+if(EXISTS ${PACKAGE_MAKER_PROGRAM})
+  list(APPEND CPACK_GENERATOR "PackageMaker")
+endif(EXISTS ${PACKAGE_MAKER_PROGRAM})
+ 
+set(CPACK_GENERATOR "${CPACK_GENERATOR}")
+set(CPACK_MONOLITHIC_INSTALL 1)
+set(CPACK_SET_DESTDIR ON)
+set(CPACK_PACKAGE_VERSION_MAJOR "@CMINPACK_MAJOR_VERSION@")
+set(CPACK_PACKAGE_VERSION_MINOR "@CMINPACK_MINOR_VERSION@")
+set(CPACK_PACKAGE_VERSION_PATCH "@CMINPACK_REVISION_VERSION@")
+include(InstallRequiredSystemLibraries)
+set(CPACK_PACKAGE_CONTACT "Frederic Devernay")
+include(CPack)
+
--- cminpack-1.1.3/CMakeLists.txt	2011-04-15 07:29:23.491254866 +0000
+++ cminpack-1.1.3/CMakeLists.txt	2011-04-15 07:30:17.260633372 +0000
@@ -5,6 +5,7 @@
 project (CMINPACK)
 string(TOLOWER ${PROJECT_NAME} PROJECT_NAME_LOWER)
 
+set(CMAKE_INSTALL_PREFIX /usr)
 include(${PROJECT_SOURCE_DIR}/cmake/cminpack_utils.cmake)
 # Set version and OS-specific settings
 set(CMINPACK_VERSION 1.0.90 CACHE STRING "CMinpack version")
