--- cminpack-1.1.3/CMakeLists.txt	2011-04-15 07:28:11.841140175 +0000
+++ cminpack-1.1.3/CMakeLists.txt	2011-04-15 07:28:38.841934851 +0000
@@ -67,3 +67,8 @@
 
 set_target_properties(cminpack PROPERTIES VERSION ${CMINPACK_VERSION})
 
+set(CPACK_GENERATOR "DEB;RPM;")
+include(InstallRequiredSystemLibraries)
+set(CPACK_PACKAGE_CONTACT "Frederic Devernay")
+include(CPack)
+
--- cminpack-1.1.3/CMakeLists.txt	2011-04-15 07:29:23.491254866 +0000
+++ cminpack-1.1.3/CMakeLists.txt	2011-04-15 07:30:17.260633372 +0000
@@ -5,6 +5,7 @@
 project (CMINPACK)
 string(TOLOWER ${PROJECT_NAME} PROJECT_NAME_LOWER)
 
+set(CMAKE_INSTALL_PREFIX /usr)
 include(${PROJECT_SOURCE_DIR}/cmake/cminpack_utils.cmake)
 # Set version and OS-specific settings
 set(CMINPACK_VERSION 1.0.90 CACHE STRING "CMinpack version")
