--- flann-1.6.8-src/cmake/flann.pc.in	2011-04-15 07:09:34.141641032 +0000
+++ flann-1.6.8-src/cmake/flann.pc.in	2011-04-15 07:10:15.141296537 +0000
@@ -8,5 +8,5 @@
 Description: @PKG_DESC@
 Version: @FLANN_VERSION@
 Requires: @PKG_EXTERNAL_DEPS@
-Libs: -L${libdir} -lflann
+Libs: -L${libdir} -lflann_cpp -lflann_cpp-gd
 Cflags: -I${includedir}

--- flann-1.6.8-src/CMakeLists.txt	2011-02-26 08:36:53.000000000 +0000
+++ flann-1.6.8-src/CMakeLists.txt	2011-04-15 06:55:12.000000000 +0000
@@ -116,3 +116,44 @@
 add_subdirectory( examples )
 add_subdirectory( test )
 add_subdirectory( doc )
+
+# RPM
+find_program(RPM_PROGRAM rpm)
+if(EXISTS ${RPM_PROGRAM})
+  list(APPEND CPACK_GENERATOR "RPM")
+endif(EXISTS ${RPM_PROGRAM})
+# DEB
+find_program(DPKG_PROGRAM dpkg)
+if(EXISTS ${DPKG_PROGRAM})
+  list(APPEND CPACK_GENERATOR "DEB")
+endif(EXISTS ${DPKG_PROGRAM})
+# NSIS
+find_program(NSIS_PROGRAM makensis MakeNSIS)
+if(EXISTS ${NSIS_PROGRAM})
+  list(APPEND CPACK_GENERATOR "NSIS")
+endif(EXISTS ${NSIS_PROGRAM})
+# dpkg
+find_program(PACKAGE_MAKER_PROGRAM PackageMaker
+	    HINTS /Developer/Applications/Utilities)
+if(EXISTS ${PACKAGE_MAKER_PROGRAM})
+  list(APPEND CPACK_GENERATOR "PackageMaker")
+endif(EXISTS ${PACKAGE_MAKER_PROGRAM})
+ 
+if(CMAKE_SIZEOF_VOID_P EQUAL 8)
+  set(CPACK_NSIS_PROGRAMFILES "$PROGRAMFILES64")
+else()
+  set(CPACK_NSIS_PROGRAMFILES "$PROGRAMFILES32")
+endif()
+set(CPACK_NSIS_INSTALL_ROOT "${CPACK_NSIS_PROGRAMFILES}")
+set(CPACK_GENERATOR "${CPACK_GENERATOR}")
+set(CPACK_MONOLITHIC_INSTALL 1)
+#set(CPACK_SET_DESTDIR ON)
+include(InstallRequiredSystemLibraries)
+set(CPACK_PACKAGE_CONTACT "Marius Muja")
+#set(CPACK_PACKAGING_INSTALL_PREFIX ${CMAKE_INSTALL_PREFIX})
+SET(CPACK_PACKAGE_VERSION ${FLANN_VERSION})
+SET(CPACK_PACKAGE_VERSION_MAJOR ${FLANN_VERSION_MAJOR})
+SET(CPACK_PACKAGE_VERSION_MINOR ${FLANN_VERSION_MINOR})
+SET(CPACK_PACKAGE_VERSION_PATCH ${FLANN_VERSION_PATCH})
+include(CPack)
+

--- flann-1.6.8-src/src/cpp/CMakeLists.txt	2011-02-26 09:36:53.000000000 +0100
+++ flann-1.6.8-src/src/cpp/CMakeLists.txt	2011-04-25 08:57:02.015164002 +0200
@@ -29,6 +30,11 @@
    SOVERSION ${FLANN_SOVERSION}
 ) 
 
+add_library(flann_cpp-gd SHARED ${CPP_SOURCES})
+set_target_properties(flann_cpp-gd PROPERTIES COMPILE_FLAGS ${CMAKE_CXX_FLAGS_DEBUG})
+add_library(flann_cpp_s-gd STATIC ${CPP_SOURCES})
+set_target_properties(flann_cpp_s-gd PROPERTIES COMPILE_FLAGS ${CMAKE_CXX_FLAGS_DEBUG})
+
 if (BUILD_C_BINDINGS)
     add_library(flann_s STATIC ${C_SOURCES})
     if(CMAKE_COMPILER_IS_GNUCC)
@@ -61,7 +67,7 @@
 
 
 install (
-    TARGETS flann_cpp flann_cpp_s
+    TARGETS flann_cpp flann_cpp_s flann_cpp-gd flann_cpp_s-gd
     RUNTIME DESTINATION bin
     LIBRARY DESTINATION ${FLANN_LIB_INSTALL_DIR}
     ARCHIVE DESTINATION ${FLANN_LIB_INSTALL_DIR}
