--- flann-1.6.8-src/cmake/flann.pc.in	2011-04-15 07:09:34.141641032 +0000
+++ flann-1.6.8-src/cmake/flann.pc.in	2011-04-15 07:10:15.141296537 +0000
@@ -8,6 +8,6 @@
 Description: @PKG_DESC@
 Version: @FLANN_VERSION@
 Requires: @PKG_EXTERNAL_DEPS@
-Libs: -L${libdir} -lflann
+Libs: -L${libdir} -lflann_cpp
 Cflags: -I${includedir}

--- flann-1.6.8-src/CMakeLists.txt	2011-02-26 08:36:53.000000000 +0000
+++ flann-1.6.8-src/CMakeLists.txt	2011-04-15 06:55:12.000000000 +0000
@@ -19,7 +19,7 @@
     "${PROJECT_BINARY_DIR}/uninstall_target.cmake")
 
 if (NOT DEFINED CMAKE_INSTALL_PREFIX)
-    set(CMAKE_INSTALL_PREFIX /usr/local)
+    set(CMAKE_INSTALL_PREFIX /usr)
 endif()
 
 # Set the build type.  Options are:
@@ -116,3 +116,38 @@
 add_subdirectory( examples )
 add_subdirectory( test )
 add_subdirectory( doc )
+
+# RPM
+find_program(RPM_PROGRAM rpm)
+if(EXISTS ${RPM_PROGRAM})
+  list(APPEND CPACK_GENERATOR "RPM")
+endif(EXISTS ${RPM_PROGRAM})
+# DEB
+find_program(DPKG_PROGRAM dpkg)
+if(EXISTS ${DPKG_PROGRAM})
+  list(APPEND CPACK_GENERATOR "DEB")
+endif(EXISTS ${DPKG_PROGRAM})
+# NSIS
+find_program(NSIS_PROGRAM makensis MakeNSIS)
+if(EXISTS ${NSIS_PROGRAM})
+  list(APPEND CPACK_GENERATOR "NSIS")
+endif(EXISTS ${NSIS_PROGRAM})
+# dpkg
+find_program(PACKAGE_MAKER_PROGRAM PackageMaker
+	    HINTS /Developer/Applications/Utilities)
+if(EXISTS ${PACKAGE_MAKER_PROGRAM})
+  list(APPEND CPACK_GENERATOR "PackageMaker")
+endif(EXISTS ${PACKAGE_MAKER_PROGRAM})
+ 
+set(CPACK_GENERATOR "${CPACK_GENERATOR}")
+set(CPACK_MONOLITHIC_INSTALL 1)
+set(CPACK_SET_DESTDIR ON)
+include(InstallRequiredSystemLibraries)
+set(CPACK_PACKAGE_CONTACT "Marius Muja")
+set(CPACK_PACKAGING_INSTALL_PREFIX ${CMAKE_INSTALL_PREFIX})
+SET(CPACK_PACKAGE_VERSION ${FLANN_VERSION})
+SET(CPACK_PACKAGE_VERSION_MAJOR ${FLANN_VERSION_MAJOR})
+SET(CPACK_PACKAGE_VERSION_MINOR ${FLANN_VERSION_MINOR})
+SET(CPACK_PACKAGE_VERSION_PATCH ${FLANN_VERSION_PATCH})
+include(CPack)
+

