Fix a memory leak and a crash in the Java bindings.

Index: OpenNI/Wrappers/OpenNI.java/src/org/OpenNI/Context.java
===================================================================
--- OpenNI.orig/Wrappers/OpenNI.java/src/org/OpenNI/Context.java	2012-09-26 14:12:06.392563448 +0400
+++ OpenNI/Wrappers/OpenNI.java/src/org/OpenNI/Context.java	2012-09-26 14:41:29.111936956 +0400
@@ -129,7 +129,9 @@
 		OutArg<Long> phNode = new OutArg<Long>();
 		int status = NativeMethods.xnCreateProductionTree(toNative(), nodeInfo.toNative(), phNode);
 		WrapperUtils.throwOnError(status);
-		return createProductionNodeObject(phNode.value);
+		ProductionNode node = createProductionNodeObject(phNode.value);
+		NativeMethods.xnProductionNodeRelease(phNode.value);
+		return node;
 	}
 	
 	public ProductionNode createAnyProductionTree(NodeType type, Query query) throws GeneralException
@@ -141,7 +143,9 @@
 				phNode,
 				errors.toNative());
 		WrapperUtils.checkEnumeration(status, errors);
-		return createProductionNodeFromNative(phNode.value);
+		ProductionNode node = createProductionNodeFromNative(phNode.value);
+		NativeMethods.xnProductionNodeRelease(phNode.value);
+		return node;
 	}
 
 	public NodeInfoList enumerateExistingNodes() throws GeneralException
@@ -378,6 +382,11 @@
 		return createProductionNodeObject(hNode, type);
 	}
 
+	void removeProductionNodeObject(long hNode)
+	{
+		this.allNodes.remove(hNode);
+	}
+
 	private Observable<ErrorStateEventArgs> errorStateChangedEvent;
 	private Hashtable<Long, ProductionNode> allNodes = new Hashtable<Long, ProductionNode>();
 	
Index: OpenNI/Wrappers/OpenNI.java/src/org/OpenNI/NodeWrapper.java
===================================================================
--- OpenNI.orig/Wrappers/OpenNI.java/src/org/OpenNI/NodeWrapper.java	2012-09-26 14:12:06.362562179 +0400
+++ OpenNI/Wrappers/OpenNI.java/src/org/OpenNI/NodeWrapper.java	2012-09-26 14:41:22.592561999 +0400
@@ -47,6 +47,7 @@
 
 	protected void freeObject(long ptr)
 	{
+		context.removeProductionNodeObject(ptr);
 		NativeMethods.xnProductionNodeRelease(ptr);
 	}
 	
Index: OpenNI/Wrappers/OpenNI.jni/org_OpenNI_NativeMethods.cpp
===================================================================
--- OpenNI.orig/Wrappers/OpenNI.jni/org_OpenNI_NativeMethods.cpp	2012-09-26 14:12:06.442561766 +0400
+++ OpenNI/Wrappers/OpenNI.jni/org_OpenNI_NativeMethods.cpp	2012-09-26 14:41:22.592561999 +0400
@@ -699,8 +699,10 @@
 	jclass cls = env->FindClass("org/OpenNI/ProductionNodeDescription");
 	jmethodID clsCtor = env->GetMethodID(cls, "<init>", "(ILjava/lang/String;Ljava/lang/String;Lorg/OpenNI/Version;)V");
 
+	jstring vendor = env->NewStringUTF(pDesc->strVendor);
+	jstring name = env->NewStringUTF(pDesc->strName);
 	jobject version = CreateVersion(env, &pDesc->Version);
-	return env->NewObject(cls, clsCtor, pDesc->Type, pDesc->strVendor, pDesc->strName, version);
+	return env->NewObject(cls, clsCtor, pDesc->Type, vendor, name, version);
 }
 
 //---------------------------------------------------------------------------
