diff -urN original_files/openni/Platform.Arm modified_files/openni/Platform.Arm
--- original_files/openni/Platform/Linux/Build/Common/Platform.Arm	2012-09-05 17:11:36.000000000 -0700
+++ modified_files/openni/Platform/Linux/Build/Common/Platform.Arm	2012-09-05 17:08:44.000000000 -0700
@@ -1,7 +1,7 @@
 ifeq "$(CFG)" "Release"
 
     # Hardware specifying flags
-    CFLAGS += -march=armv7-a -mtune=cortex-a8 -mfpu=neon -mfloat-abi=softfp #-mcpu=cortex-a8
+    CFLAGS += -march=armv7-a -mtune=cortex-a8 -mfpu=neon  #-mcpu=cortex-a8
 
     # Optimization level, minus currently buggy optimizing methods (which break bit-exact)
     CFLAGS += -O3 -fno-tree-pre -fno-strict-aliasing
diff -urN original_files/openni/Redist_OpenNi.py modified_files/openni/Redist_OpenNi.py
--- original_files/openni/Platform/Linux/CreateRedist/Redist_OpenNi.py	2012-09-05 17:05:30.000000000 -0700
+++ modified_files/openni/Platform/Linux/CreateRedist/Redist_OpenNi.py	2012-09-05 17:08:44.000000000 -0700
@@ -183,7 +183,7 @@
         PLATFORM = "x64"
     elif machinetype == "x86" or machinetype == "i386" or machinetype == "i686":
         PLATFORM = "x86"
-    elif machinetype == "arm":
+    elif machinetype == "armv7l":
         PLATFORM = "Arm"
     else:
         print "Unknown platform:", machinetype

