diff --git a/boost/detail/endian.hpp b/boost/detail/endian.hpp
index 98c870c..5ae0b7e 100644
--- a/boost/detail/endian.hpp
+++ b/boost/detail/endian.hpp
@@ -31,7 +31,7 @@
 // GNU libc offers the helpful header <endian.h> which defines
 // __BYTE_ORDER
 
-#if defined (__GLIBC__)
+#if defined (__GLIBC__) || defined (__ANDROID__)
 # include <endian.h>
 # if (__BYTE_ORDER == __LITTLE_ENDIAN)
 #  define BOOST_LITTLE_ENDIAN
diff --git a/boost/thread/pthread/thread_data.hpp b/boost/thread/pthread/thread_data.hpp
index 5f84799..af0aa32 100644
--- a/boost/thread/pthread/thread_data.hpp
+++ b/boost/thread/pthread/thread_data.hpp
@@ -22,6 +22,12 @@
 #endif
 #include <boost/config/abi_prefix.hpp>
 
+// This will no longer be necessary in NDK 9.
+#ifdef __ANDROID__
+#include <asm/page.h>
+#endif
+
+
 namespace boost
 {
     class thread_attributes {
diff --git a/libs/filesystem/src/operations.cpp b/libs/filesystem/src/operations.cpp
index 16a336f..4f0c0de 100644
--- a/libs/filesystem/src/operations.cpp
+++ b/libs/filesystem/src/operations.cpp
@@ -73,7 +73,7 @@ using std::wstring;
     const fs::path dot_dot_path("..");
 #   include <sys/types.h>
 #   include <sys/stat.h>
-#   if !defined(__APPLE__) && !defined(__OpenBSD__)
+#   if !defined(__APPLE__) && !defined(__OpenBSD__) && !defined(__ANDROID__)
 #     include <sys/statvfs.h>
 #     define BOOST_STATVFS statvfs
 #     define BOOST_STATVFS_F_FRSIZE vfs.f_frsize
@@ -81,6 +81,9 @@ using std::wstring;
 #     ifdef __OpenBSD__
 #     include <sys/param.h>
 #     endif
+#     ifdef __ANDROID__
+#     include <sys/vfs.h>
+#     endif
 #     include <sys/mount.h>
 #     define BOOST_STATVFS statfs
 #     define BOOST_STATVFS_F_FRSIZE static_cast<boost::uintmax_t>(vfs.f_bsize)
diff --git a/tools/build/v2/user-config.jam b/tools/build/v2/user-config.jam
index fbbf13f..f6c07e4 100644
--- a/tools/build/v2/user-config.jam
+++ b/tools/build/v2/user-config.jam
@@ -44,6 +44,7 @@
 # Configure specific gcc version, giving alternative name to use.
 # using gcc : 3.2 : g++-3.2 ;
 
+using gcc : androidarmv7a : arm-linux-androideabi-g++ : <compileflags>"-march=armv7-a -mfloat-abi=softfp" <linkflags>"-Wl,--fix-cortex-a8 -Wl,--no-undefined" ;
 
 # -------------------
 # MSVC configuration.
