diff -ru VTK/Charts/CMakeLists.txt VTK/Charts/CMakeLists.txt
--- VTK/Charts/CMakeLists.txt	2010-09-27 13:40:00.000000000 -0700
+++ VTK/Charts/CMakeLists.txt	2012-02-23 20:47:40.000000000 -0800
@@ -10,9 +10,13 @@
 ENDIF (JAVA_AWT_LIBRARY)
 
 SET(KIT_INTERFACE_LIBRARIES vtkHybrid vtkViews)
+SET(KIT_INTERFACE_LIBRARIES_DEBUG vtkHybrid-gd vtkViews-gd)
 SET(KIT_LIBS vtkIO vtkftgl
   ${VTK_FREETYPE_LIBRARIES}
 )
+SET(KIT_LIBS_DEBUG vtkIO-gd vtkftgl-gd
+  ${VTK_FREETYPE_LIBRARIES_DEBUG}
+)
 
 SET(Kit_SRCS
   vtkAbstractContextBufferId.cxx
@@ -69,8 +73,10 @@
   INCLUDE(${QT_USE_FILE})
   IF (APPLE)
     LIST (APPEND KIT_INTERFACE_LIBRARIES ${QT_LIBRARIES})
+    LIST (APPEND KIT_INTERFACE_LIBRARIES_DEBUG ${QT_LIBRARIES})
   ELSE (APPLE)
     LIST (APPEND KIT_LIBS ${QT_LIBRARIES})
+    LIST (APPEND KIT_LIBS_DEBUG ${QT_LIBRARIES})
   ENDIF (APPLE)
 ENDIF(VTK_USE_QT)
 
diff -ru VTK/CMake/KitCommonBlock.cmake VTK/CMake/KitCommonBlock.cmake
--- VTK/CMake/KitCommonBlock.cmake	2010-09-27 13:40:00.000000000 -0700
+++ VTK/CMake/KitCommonBlock.cmake	2012-02-23 20:47:40.000000000 -0800
@@ -5,7 +5,9 @@
                        ${VTK_BINARY_DIR} "")
 
 ADD_LIBRARY(vtk${KIT} ${Kit_SRCS} ${Kit_EXTRA_SRCS} ${KitInstantiator_SRCS})
-SET(KIT_LIBRARY_TARGETS ${KIT_LIBRARY_TARGETS} vtk${KIT})
+ADD_LIBRARY(vtk${KIT}-gd ${Kit_SRCS} ${Kit_EXTRA_SRCS} ${KitInstantiator_SRCS})
+SET_TARGET_PROPERTIES(vtk${KIT}-gd PROPERTIES COMPILE_FLAGS ${CMAKE_CXX_FLAGS_DEBUG})
+SET(KIT_LIBRARY_TARGETS ${KIT_LIBRARY_TARGETS} vtk${KIT} vtk${KIT}-gd)
 # Add target specific compile flags for ABI setttings
 IF(VTK_ABI_CXX_FLAGS)
   SET_PROPERTY(TARGET vtk${KIT} APPEND 
@@ -40,12 +42,17 @@
 ENDIF (VTK_WRAP_JAVA)
 
 TARGET_LINK_LIBRARIES(vtk${KIT} ${KIT_INTERFACE_LIBRARIES} ${KIT_LIBS})
+TARGET_LINK_LIBRARIES(vtk${KIT}-gd ${KIT_INTERFACE_LIBRARIES_DEBUG} ${KIT_LIBS_DEBUG})
 
 IF(NOT VTK_INSTALL_NO_LIBRARIES)
   INSTALL(TARGETS vtk${KIT} 
     RUNTIME DESTINATION ${VTK_INSTALL_BIN_DIR_CM24} COMPONENT RuntimeLibraries
     LIBRARY DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT RuntimeLibraries
     ARCHIVE DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT Development)
+  INSTALL(TARGETS vtk${KIT}-gd 
+    RUNTIME DESTINATION ${VTK_INSTALL_BIN_DIR_CM24} COMPONENT RuntimeLibraries
+    LIBRARY DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT RuntimeLibraries
+    ARCHIVE DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT Development)
 ENDIF(NOT VTK_INSTALL_NO_LIBRARIES)
 IF(NOT VTK_INSTALL_NO_DEVELOPMENT)
   SET(__files.h)
@@ -83,4 +90,5 @@
 # Mark KIT_INTERFACE_LIBRARIES as LINK_INTERFACE_LIBRARIES such that any executable
 # or library that links vtk${KIT} also links against KIT_INTERFACE_LIBRARIES
 TARGET_LINK_LIBRARIES(vtk${KIT} LINK_INTERFACE_LIBRARIES ${KIT_INTERFACE_LIBRARIES})
+TARGET_LINK_LIBRARIES(vtk${KIT}-gd LINK_INTERFACE_LIBRARIES ${KIT_INTERFACE_LIBRARIES_DEBUG})
 
diff -ru VTK/CMake/vtkThirdParty.cmake VTK/CMake/vtkThirdParty.cmake
--- VTK/CMake/vtkThirdParty.cmake	2010-09-27 13:40:00.000000000 -0700
+++ VTK/CMake/vtkThirdParty.cmake	2012-02-23 20:47:40.000000000 -0800
@@ -20,6 +20,7 @@
     ENDIF(${upper}_FOUND)
   ELSE(VTK_USE_SYSTEM_${upper})
     SET(VTK_${upper}_LIBRARIES vtk${lower})
+    SET(VTK_${upper}_LIBRARIES_DEBUG vtk${lower}-gd)
   ENDIF(VTK_USE_SYSTEM_${upper})
 ENDMACRO(VTK_THIRD_PARTY_OPTION)
 
diff -ru VTK/Common/CMakeLists.txt VTK/Common/CMakeLists.txt
--- VTK/Common/CMakeLists.txt	2010-09-27 13:40:00.000000000 -0700
+++ VTK/Common/CMakeLists.txt	2012-02-23 20:47:40.000000000 -0800
@@ -4,6 +4,7 @@
 SET(KIT_PYTHON_LIBS)
 SET(KIT_JAVA_LIBS)
 SET(KIT_INTERFACE_LIBRARIES ${KWSYS_NAMESPACE})
+SET(KIT_INTERFACE_LIBRARIES_DEBUG ${KWSYS_NAMESPACE}-gd)
 
 IF(WIN32)
   IF (NOT VTK_USE_X)
@@ -14,7 +15,9 @@
 
 IF (UNIX)
   SET(KIT_LIBS ${KIT_LIBS} ${CMAKE_THREAD_LIBS} ${CMAKE_DL_LIBS})
+	SET(KIT_LIBS_DEBUG ${KIT_LIBS_DEBUG} ${CMAKE_THREAD_LIBS} ${CMAKE_DL_LIBS})
   SET(KIT_INTERFACE_LIBRARIES ${KWSYS_NAMESPACE} -lm)
+  SET(KIT_INTERFACE_LIBRARIES_DEBUG ${KWSYS_NAMESPACE}-gd -lm)
 ENDIF (UNIX)
 
 SET( Kit_SRCS
@@ -457,6 +460,7 @@
 IF (WIN32)
   IF (NOT BORLAND)
     SET(KIT_LIBS ${KIT_LIBS} wsock32)
+		SET(KIT_LIBS_DEBUG ${KIT_LIBS_DEBUG} wsock32)
   ENDIF (NOT BORLAND)
 ENDIF (WIN32)
 
@@ -470,6 +474,7 @@
 # and socket also
 IF(CMAKE_SYSTEM MATCHES "SunOS.*")
   SET(KIT_LIBS ${KIT_LIBS} socket nsl)
+	SET(KIT_LIBS_DEBUG ${KIT_LIBS_DEBUG} socket nsl)
 ENDIF(CMAKE_SYSTEM MATCHES "SunOS.*") 
 
 #-----------------------------------------------------------------------------
diff -ru VTK/Examples/Build/vtkLocal/CMakeLists.txt VTK/Examples/Build/vtkLocal/CMakeLists.txt
--- VTK/Examples/Build/vtkLocal/CMakeLists.txt	2010-09-27 13:40:01.000000000 -0700
+++ VTK/Examples/Build/vtkLocal/CMakeLists.txt	2012-02-23 20:47:41.000000000 -0800
@@ -87,6 +87,9 @@
   # Create the library.
   ADD_LIBRARY(${PROJECT_NAME} ${vtkLocal_SRCS} ${vtkLocalInstantiator_SRCS})
   TARGET_LINK_LIBRARIES(${PROJECT_NAME} ${vtkLocal_LIBS})
+  ADD_LIBRARY(${PROJECT_NAME}-gd ${vtkLocal_SRCS} ${vtkLocalInstantiator_SRCS})
+  TARGET_LINK_LIBRARIES(${PROJECT_NAME}-gd ${vtkLocal_LIBS})
+	SET_TARGET_PROPERTIES(${PROJECT_NAME}-gd PROPERTIES COMPILE_FLAGS ${CMAKE_CXX_FLAGS_DEBUG})
 
   # Create Tcl wrappers if VTK is wrapped in Tcl.
   IF(VTK_WRAP_TCL)
diff -ru VTK/Examples/Build/vtkMy/Common/CMakeLists.txt VTK/Examples/Build/vtkMy/Common/CMakeLists.txt
--- VTK/Examples/Build/vtkMy/Common/CMakeLists.txt	2010-09-27 13:40:01.000000000 -0700
+++ VTK/Examples/Build/vtkMy/Common/CMakeLists.txt	2012-02-23 20:47:41.000000000 -0800
@@ -14,6 +14,9 @@
 # Create the vtkmyCommon C++ library.
 ADD_LIBRARY (vtkmyCommon ${Common_SRCS})
 TARGET_LINK_LIBRARIES (vtkmyCommon vtkCommon)
+ADD_LIBRARY (vtkmyCommon-gd ${Common_SRCS})
+SET_TARGET_PROPERTIES (vtkmyCommon-gd PROPERTIES COMPILE_FLAGS ${CMAKE_CXX_FLAGS_DEBUG})
+TARGET_LINK_LIBRARIES (vtkmyCommon-gd vtkCommon)
 
 # Create the vtkmyCommon Tcl library.
 IF (VTK_WRAP_TCL AND VTKMY_WRAP_TCL)
diff -ru VTK/Examples/Build/vtkMy/Imaging/CMakeLists.txt VTK/Examples/Build/vtkMy/Imaging/CMakeLists.txt
--- VTK/Examples/Build/vtkMy/Imaging/CMakeLists.txt	2010-09-27 13:40:01.000000000 -0700
+++ VTK/Examples/Build/vtkMy/Imaging/CMakeLists.txt	2012-02-23 20:47:41.000000000 -0800
@@ -14,6 +14,9 @@
 # Create the vtkmyImaging C++ library
 ADD_LIBRARY (vtkmyImaging ${Imaging_SRCS})
 TARGET_LINK_LIBRARIES(vtkmyImaging vtkmyCommon vtkImaging)
+ADD_LIBRARY (vtkmyImaging-gd ${Imaging_SRCS})
+TARGET_LINK_LIBRARIES(vtkmyImaging-gd vtkmyCommon-gd vtkImaging-gd)
+SET_TARGET_PROPERTIES(vtkmyImaging-gd PROPERTIES COMPILE_FLAGS ${CMAKE_CXX_FLAGS_DEBUG})
 
 # Create the vtkmyImaging Tcl library
 IF (VTK_WRAP_TCL AND VTKMY_WRAP_TCL)
diff -ru VTK/Examples/Build/vtkMy/Unsorted/CMakeLists.txt VTK/Examples/Build/vtkMy/Unsorted/CMakeLists.txt
--- VTK/Examples/Build/vtkMy/Unsorted/CMakeLists.txt	2010-09-27 13:40:01.000000000 -0700
+++ VTK/Examples/Build/vtkMy/Unsorted/CMakeLists.txt	2012-02-23 20:47:41.000000000 -0800
@@ -14,9 +14,13 @@
 # Create the vtkmyUnsorted C++ library.
 ADD_LIBRARY (vtkmyUnsorted ${Unsorted_SRCS})
 TARGET_LINK_LIBRARIES(vtkmyUnsorted vtkmyImaging vtkGraphics vtkIO)
+ADD_LIBRARY (vtkmyUnsorted-gd ${Unsorted_SRCS})
+TARGET_LINK_LIBRARIES(vtkmyUnsorted-gd vtkmyImaging vtkGraphics vtkIO)
+SET_TARGET_PROPERTIES(vtkmyUnsorted-gd PROPERTIES COMPILE_FLAGS ${CMAKE_CXX_FLAGS_DEBUG})
 
 IF (VTK_USE_PARALLEL)
   TARGET_LINK_LIBRARIES (vtkmyUnsorted vtkParallel)
+	TARGET_LINK_LIBRARIES (vtkmyUnsorted-gd vtkParallel-gd)
 ENDIF (VTK_USE_PARALLEL)
 
 IF (VTK_USE_RENDERING)
@@ -24,6 +28,10 @@
   TARGET_LINK_LIBRARIES (vtkmyUnsorted vtkVolumeRendering)
   TARGET_LINK_LIBRARIES (vtkmyUnsorted vtkHybrid)
   TARGET_LINK_LIBRARIES (vtkmyUnsorted vtkWidgets)
+  TARGET_LINK_LIBRARIES (vtkmyUnsorted-gd vtkRendering-gd)
+  TARGET_LINK_LIBRARIES (vtkmyUnsorted-gd vtkVolumeRendering-gd)
+  TARGET_LINK_LIBRARIES (vtkmyUnsorted-gd vtkHybrid-gd)
+  TARGET_LINK_LIBRARIES (vtkmyUnsorted-gd vtkWidgets-gd)
 ENDIF (VTK_USE_RENDERING)
 
 # Create the vtkmyUnsorted Tcl library.
diff -ru VTK/Filtering/CMakeLists.txt VTK/Filtering/CMakeLists.txt
--- VTK/Filtering/CMakeLists.txt	2010-09-27 13:40:02.000000000 -0700
+++ VTK/Filtering/CMakeLists.txt	2012-02-23 20:47:42.000000000 -0800
@@ -4,6 +4,7 @@
 SET(KIT_PYTHON_LIBS vtkCommonPythonD)
 SET(KIT_JAVA_LIBS vtkCommonJava)
 SET(KIT_INTERFACE_LIBRARIES vtkCommon)
+SET(KIT_INTERFACE_LIBRARIES_DEBUG vtkCommon-gd)
 
 SET( Kit_SRCS
 vtkAbstractCellLocator.cxx
diff -ru VTK/GenericFiltering/CMakeLists.txt VTK/GenericFiltering/CMakeLists.txt
--- VTK/GenericFiltering/CMakeLists.txt	2010-09-27 13:40:03.000000000 -0700
+++ VTK/GenericFiltering/CMakeLists.txt	2012-02-23 20:47:42.000000000 -0800
@@ -4,6 +4,7 @@
 SET(KIT_PYTHON_LIBS vtkFilteringPythonD vtkGraphicsPythonD)
 SET(KIT_JAVA_LIBS vtkFilteringJava vtkGraphicsJava)
 SET(KIT_INTERFACE_LIBRARIES vtkFiltering vtkGraphics)
+SET(KIT_INTERFACE_LIBRARIES_DEBUG vtkFiltering-gd vtkGraphics-gd)
 
 SET( Kit_SRCS
 vtkGenericContourFilter.cxx
diff -ru VTK/Geovis/CMakeLists.txt VTK/Geovis/CMakeLists.txt
--- VTK/Geovis/CMakeLists.txt	2010-09-27 13:40:03.000000000 -0700
+++ VTK/Geovis/CMakeLists.txt	2012-02-23 20:47:42.000000000 -0800
@@ -4,7 +4,9 @@
 SET(KIT_PYTHON_LIBS vtkWidgetsPythonD vtkViewsPythonD)
 SET(KIT_JAVA_LIBS vtkWidgetsJava vtkViewsJava)
 SET(KIT_INTERFACE_LIBRARIES vtkWidgets vtkViews)
+SET(KIT_INTERFACE_LIBRARIES_DEBUG vtkWidgets-gd vtkViews-gd)
 SET(KIT_LIBS ${VTK_LIBPROJ4_LIBRARIES} ${OPENGL_LIBRARIES})
+SET(KIT_LIBS_DEBUG ${VTK_LIBPROJ4_LIBRARIES}-gd ${OPENGL_LIBRARIES})
 
 SET( Kit_SRCS
   vtkCompassRepresentation.cxx
diff -ru VTK/Graphics/CMakeLists.txt VTK/Graphics/CMakeLists.txt
--- VTK/Graphics/CMakeLists.txt	2010-09-27 13:40:03.000000000 -0700
+++ VTK/Graphics/CMakeLists.txt	2012-02-23 20:47:42.000000000 -0800
@@ -7,7 +7,9 @@
   SET(KIT_JAVA_LIBS ${KIT_JAVA_LIBS} ${JAVA_AWT_LIBRARY} ${JAVA_JVM_LIBRARY})
 ENDIF(JAVA_AWT_LIBRARY)
 SET(KIT_INTERFACE_LIBRARIES vtkFiltering)
+SET(KIT_INTERFACE_LIBRARIES_DEBUG vtkFiltering-gd)
 SET(KIT_LIBS vtkverdict )
+SET(KIT_LIBS_DEBUG vtkverdict-gd )
 SET( Kit_SRCS
 vtkAnnotationLink.cxx
 vtkAppendCompositeDataLeaves.cxx
@@ -300,6 +302,7 @@
     WRAP_EXCLUDE
   )
   SET(KIT_LIBS ${KIT_LIBS} ${MEX_LIBRARIES} ${MATLAB_ENGINE_LIBRARIES})
+	SET(KIT_LIBS_DEBUG ${KIT_LIBS_DEBUG} ${MEX_LIBRARIES} ${MATLAB_ENGINE_LIBRARIES})
 ENDIF(VTK_USE_MATLAB_MEX)
 
 # Add Gnu R interface related files.
@@ -317,6 +320,7 @@
   )
 
   SET(KIT_LIBS ${KIT_LIBS} ${R_LIBRARIES})
+	SET(KIT_LIBS_DEBUG ${KIT_LIBS_DEBUG} ${R_LIBRARIES})
 
 ENDIF(VTK_USE_GNU_R)
 
diff -ru VTK/GUISupport/MFC/CMakeLists.txt VTK/GUISupport/MFC/CMakeLists.txt
--- VTK/GUISupport/MFC/CMakeLists.txt	2010-09-27 13:40:03.000000000 -0700
+++ VTK/GUISupport/MFC/CMakeLists.txt	2012-02-23 20:47:44.000000000 -0800
@@ -18,14 +18,18 @@
   ) 
 
 ADD_LIBRARY(vtkMFC ${VTK_MFC_LIB_TYPE} ${MFC_SRCS})
+ADD_LIBRARY(vtkMFC-gd ${VTK_MFC_LIB_TYPE} ${MFC_SRCS})
+SET_TARGET_PROPERTIES(vtkMFC-gd PROPERTIES COMPILE_FLAGS ${CMAKE_CXX_FLAGS_DEBUG})
 
 # Apply user-defined properties to the library target.
 IF(VTK_LIBRARY_PROPERTIES)
   SET_TARGET_PROPERTIES(vtkMFC PROPERTIES ${VTK_LIBRARY_PROPERTIES})
+  SET_TARGET_PROPERTIES(vtkMFC-gd PROPERTIES ${VTK_LIBRARY_PROPERTIES})
 ENDIF(VTK_LIBRARY_PROPERTIES)
 
 IF(VTK_MFC_EXTRA_LIBS)
   TARGET_LINK_LIBRARIES(vtkMFC ${VTK_MFC_EXTRA_LIBS})
+  TARGET_LINK_LIBRARIES(vtkMFC-gd ${VTK_MFC_EXTRA_LIBS})
 ENDIF(VTK_MFC_EXTRA_LIBS)
 
 TARGET_LINK_LIBRARIES(vtkMFC vtkRendering)
@@ -45,6 +49,11 @@
       RUNTIME DESTINATION ${VTK_INSTALL_BIN_DIR_CM24} COMPONENT RuntimeLibraries # .exe, .dll
       LIBRARY DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT RuntimeLibraries # .so, mod.dll
       ARCHIVE DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT Development      # .a, .lib
+      )
+    INSTALL(TARGETS vtkMFC-gd
+      RUNTIME DESTINATION ${VTK_INSTALL_BIN_DIR_CM24} COMPONENT RuntimeLibraries # .exe, .dll
+      LIBRARY DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT RuntimeLibraries # .so, mod.dll
+      ARCHIVE DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT Development      # .a, .lib
       )
   ENDIF(NOT VTK_INSTALL_NO_LIBRARIES)
 ENDIF ( WIN32 )
diff -ru VTK/GUISupport/Qt/Chart/CMakeLists.txt VTK/GUISupport/Qt/Chart/CMakeLists.txt
--- VTK/GUISupport/Qt/Chart/CMakeLists.txt	2010-09-27 13:40:03.000000000 -0700
+++ VTK/GUISupport/Qt/Chart/CMakeLists.txt	2012-02-23 20:47:44.000000000 -0800
@@ -306,14 +306,25 @@
     ${vtkQtChart_MOC_SOURCES}
   )
 
+  ADD_LIBRARY(vtkQtChart-gd
+    ${vtkQtChart_SOURCES}
+    ${vtkQtChart_HEADERS}
+    ${vtkQtChart_MOC_SOURCES}
+  )
+  SET_TARGET_PROPERTIES(vtkQtChart-gd PROPERTIES COMPILE_FLAGS ${CMAKE_CXX_FLAGS_DEBUG})
+
   # Apply user-defined properties to the library target.
   IF(VTK_LIBRARY_PROPERTIES)
     SET_TARGET_PROPERTIES(vtkQtChart PROPERTIES ${VTK_LIBRARY_PROPERTIES})
+    SET_TARGET_PROPERTIES(vtkQtChart-gd PROPERTIES ${VTK_LIBRARY_PROPERTIES})
   ENDIF(VTK_LIBRARY_PROPERTIES)
 
   TARGET_LINK_LIBRARIES(vtkQtChart
     ${QT_LIBRARIES}
   )
+  TARGET_LINK_LIBRARIES(vtkQtChart-gd
+    ${QT_LIBRARIES}
+  )
 
   IF(NOT VTK_INSTALL_NO_DEVELOPMENT AND VTK_INSTALL_INCLUDE_DIR_CM24)
     INSTALL(FILES ${vtkQtChart_MOCHEADERS} ${vtkQtChart_HEADERS}
@@ -325,6 +336,10 @@
       RUNTIME DESTINATION ${VTK_INSTALL_BIN_DIR_CM24} COMPONENT RuntimeLibraries
       LIBRARY DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT RuntimeLibraries
       ARCHIVE DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT Development)
+    INSTALL(TARGETS vtkQtChart-gd
+      RUNTIME DESTINATION ${VTK_INSTALL_BIN_DIR_CM24} COMPONENT RuntimeLibraries
+      LIBRARY DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT RuntimeLibraries
+      ARCHIVE DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT Development)
   ENDIF(NOT VTK_INSTALL_NO_LIBRARIES AND VTK_INSTALL_BIN_DIR_CM24 AND VTK_INSTALL_LIB_DIR_CM24)
 
 ENDIF(NOT QT4_FOUND)
diff -ru VTK/GUISupport/Qt/CMakeLists.txt VTK/GUISupport/Qt/CMakeLists.txt
--- VTK/GUISupport/Qt/CMakeLists.txt	2010-09-27 13:40:03.000000000 -0700
+++ VTK/GUISupport/Qt/CMakeLists.txt	2012-02-23 20:47:44.000000000 -0800
@@ -125,10 +125,13 @@
 SET_DIRECTORY_PROPERTIES(PROPERTIES INCLUDE_DIRECTORIES "${include_dirs_tmp}")
 
 ADD_LIBRARY(QVTK ${QVTKLibSrcs} ${QVTKLibMocSrcs})
+ADD_LIBRARY(QVTK-gd ${QVTKLibSrcs} ${QVTKLibMocSrcs})
+SET_TARGET_PROPERTIES(QVTK-gd PROPERTIES COMPILE_FLAGS ${CMAKE_CXX_FLAGS_DEBUG})
 
 # Apply user-defined properties to the library target.
 IF(VTK_LIBRARY_PROPERTIES)
   SET_TARGET_PROPERTIES(QVTK PROPERTIES ${VTK_LIBRARY_PROPERTIES})
+  SET_TARGET_PROPERTIES(QVTK-gd PROPERTIES ${VTK_LIBRARY_PROPERTIES})
 ENDIF(VTK_LIBRARY_PROPERTIES)
 
 TARGET_LINK_LIBRARIES( QVTK 
@@ -137,16 +140,25 @@
   vtkGraphics
   vtkImaging 
   vtkCommon)
+TARGET_LINK_LIBRARIES( QVTK-gd 
+  ${QT_LIBRARIES}
+  vtkRendering-gd 
+  vtkGraphics-gd
+  vtkImaging-gd 
+  vtkCommon-gd)
 
 # add extra dependencies if necessary
 TARGET_LINK_LIBRARIES( QVTK vtkQtChart )
+TARGET_LINK_LIBRARIES( QVTK-gd vtkQtChart-gd )
 IF(VTK_USE_VIEWS)
   TARGET_LINK_LIBRARIES( QVTK vtkViews )
+  TARGET_LINK_LIBRARIES( QVTK-gd vtkViews-gd )
 ENDIF(VTK_USE_VIEWS)
 
 IF(APPLE)
   IF(VTK_USE_CARBON)
     TARGET_LINK_LIBRARIES( QVTK "-framework Carbon" )
+    TARGET_LINK_LIBRARIES( QVTK-gd "-framework Carbon" )
   ENDIF(VTK_USE_CARBON)
 ENDIF(APPLE)
 
@@ -174,12 +186,28 @@
   ${PluginLibSrcs}
   ${PluginMocSrcs}
 )
+ADD_LIBRARY ( QVTKWidgetPlugin-gd
+  SHARED
+  ${PluginLibSrcs}
+  ${PluginMocSrcs}
+)
 
 # link with release version of Qt libs
 TARGET_LINK_LIBRARIES( QVTKWidgetPlugin 
   ${QT_PLUGIN_LIBS}
 )
 
+ADD_LIBRARY ( QVTKWidgetPlugin-gd
+  SHARED
+  ${PluginLibSrcs}
+  ${PluginMocSrcs}
+)
+
+# link with release version of Qt libs
+TARGET_LINK_LIBRARIES( QVTKWidgetPlugin-gd
+  ${QT_PLUGIN_LIBS}
+)
+SET_TARGET_PROPERTIES( QVTKWidgetPlugin-gd PROPERTIES COMPILE_FLAGS ${CMAKE_CXX_FLAGS_DEBUG})
 
 # Configure the VTKConfigQt.cmake support file.
 CONFIGURE_FILE(${CMAKE_CURRENT_SOURCE_DIR}/VTKConfigQt.cmake.in
@@ -222,6 +250,9 @@
   SET_TARGET_PROPERTIES(QVTK PROPERTIES POST_INSTALL_SCRIPT
     ${CMAKE_CURRENT_BINARY_DIR}/PluginInstall.cmake
     )
+  SET_TARGET_PROPERTIES(QVTK-gd PROPERTIES POST_INSTALL_SCRIPT
+    ${CMAKE_CURRENT_BINARY_DIR}/PluginInstall.cmake
+    )
 ENDIF(NOT VTK_INSTALL_NO_QT_PLUGIN)
 
 IF(NOT VTK_INSTALL_NO_DEVELOPMENT)
@@ -244,6 +275,10 @@
     RUNTIME DESTINATION ${VTK_INSTALL_BIN_DIR_CM24} COMPONENT RuntimeLibraries
     LIBRARY DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT RuntimeLibraries
     ARCHIVE DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT Development)
+  INSTALL(TARGETS QVTK-gd
+    RUNTIME DESTINATION ${VTK_INSTALL_BIN_DIR_CM24} COMPONENT RuntimeLibraries
+    LIBRARY DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT RuntimeLibraries
+    ARCHIVE DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT Development)
 ENDIF(NOT VTK_INSTALL_NO_LIBRARIES)
 
 CONFIGURE_FILE("${CMAKE_CURRENT_SOURCE_DIR}/CTestCustom.ctest.in"
diff -ru VTK/Hybrid/CMakeLists.txt VTK/Hybrid/CMakeLists.txt
--- VTK/Hybrid/CMakeLists.txt	2010-09-27 13:40:04.000000000 -0700
+++ VTK/Hybrid/CMakeLists.txt	2012-02-23 20:47:44.000000000 -0800
@@ -5,8 +5,10 @@
 SET(KIT_PYTHON_LIBS vtkRenderingPythonD vtkIOPythonD)
 SET(KIT_JAVA_LIBS vtkRenderingJava vtkIOJava)
 SET(KIT_INTERFACE_LIBRARIES vtkRendering vtkIO)
+SET(KIT_INTERFACE_LIBRARIES_DEBUG vtkRendering-gd vtkIO-gd)
 IF(VTK_HAS_EXODUS)
   SET(KIT_LIBS ${KIT_LIBS} vtkexoIIc)
+  SET(KIT_LIBS_DEBUG ${KIT_LIBS_DEBUG} vtkexoIIc-gd)
 ENDIF(VTK_HAS_EXODUS)
 
 IF (WIN32)
@@ -19,14 +21,17 @@
       ENDIF(NOT HAVE_VFW32)
     ENDIF(BORLAND)
     SET(KIT_LIBS ${KIT_LIBS} vfw32)
+    SET(KIT_LIBS_DEBUG ${KIT_LIBS_DEBUG} vfw32)
   ENDIF (VTK_VFW_SUPPORTS_CAPTURE)
   IF (VTK_USE_MATROX_IMAGING)
     SET(KIT_LIBS ${KIT_LIBS} ${MIL_LIBRARY})
+    SET(KIT_LIBS_DEBUG ${KIT_LIBS_DEBUG} ${MIL_LIBRARY})
   ENDIF (VTK_USE_MATROX_IMAGING)
 ENDIF (WIN32)
 
 # for vtkImageDataLIC2D and vtkStructuredGridLIC2D
 SET( KIT_LIBS ${KIT_LIBS} vtkftgl )
+SET( KIT_LIBS_DEBUG ${KIT_LIBS_DEBUG} vtkftgl-gd )
 
 SET ( Kit_SRCS
 vtk3DSImporter.cxx
@@ -101,6 +106,7 @@
       vtkPExodusReader.cxx
       )
     SET(KIT_INTERFACE_LIBRARIES ${KIT_INTERFACE_LIBRARIES} vtkParallel)
+    SET(KIT_INTERFACE_LIBRARIES_DEBUG ${KIT_INTERFACE_LIBRARIES_DEBUG} vtkParallel-gd)
   ENDIF(VTK_USE_PARALLEL)
 ENDIF(VTK_HAS_EXODUS)
 
diff -ru VTK/Imaging/CMakeLists.txt VTK/Imaging/CMakeLists.txt
--- VTK/Imaging/CMakeLists.txt	2010-09-27 13:40:05.000000000 -0700
+++ VTK/Imaging/CMakeLists.txt	2012-02-23 20:47:44.000000000 -0800
@@ -4,6 +4,7 @@
 SET(KIT_PYTHON_LIBS vtkFilteringPythonD)
 SET(KIT_JAVA_LIBS vtkFilteringJava)
 SET(KIT_INTERFACE_LIBRARIES vtkFiltering)
+SET(KIT_INTERFACE_LIBRARIES_DEBUG vtkFiltering-gd)
 
 SET( Kit_SRCS
 vtkBooleanTexture.cxx
diff -ru VTK/Infovis/CMakeLists.txt VTK/Infovis/CMakeLists.txt
--- VTK/Infovis/CMakeLists.txt	2010-09-27 13:40:05.000000000 -0700
+++ VTK/Infovis/CMakeLists.txt	2012-02-23 20:47:45.000000000 -0800
@@ -9,13 +9,17 @@
   SET(KIT_PYTHON_LIBS vtkWidgetsPythonD)
   SET(KIT_JAVA_LIBS vtkWidgetsJava)
   SET(KIT_INTERFACE_LIBRARIES vtkWidgets)
+  SET(KIT_INTERFACE_LIBRARIES_DEBUG vtkWidgets-gd)
   SET(KIT_LIBS ${VTK_LIBXML2_LIBRARIES} vtkalglib)
+	SET(KIT_LIBS_DEBUG ${VTK_LIBXML2_LIBRARIES}-gd vtkalglib-gd)
 ELSE(VTK_USE_RENDERING)
   SET(KIT_TCL_LIBS vtkGraphicsTCL)
   SET(KIT_PYTHON_LIBS vtkGraphicsPythonD)
   SET(KIT_JAVA_LIBS vtkGraphicsJava)
   SET(KIT_INTERFACE_LIBRARIES vtkGraphics)
+  SET(KIT_INTERFACE_LIBRARIES_DEBUG vtkGraphics-gd)
   SET(KIT_LIBS ${VTK_LIBXML2_LIBRARIES} vtkalglib vtkIO vtkImaging)
+  SET(KIT_LIBS_DEBUG ${VTK_LIBXML2_LIBRARIES}-gd vtkalglib-gd vtkIO-gd vtkImaging-gd)
 ENDIF(VTK_USE_RENDERING)
 
 SET( Kit_SRCS
@@ -226,6 +230,7 @@
     ENDIF(VTK_USE_N_WAY_ARRAYS)
 
   SET(KIT_INTERFACE_LIBRARIES ${KIT_INTERFACE_LIBRARIES} vtkParallel)
+  SET(KIT_INTERFACE_LIBRARIES_DEBUG ${KIT_INTERFACE_LIBRARIES_DEBUG} vtkParallel-gd)
 ENDIF(VTK_USE_PARALLEL)
 
 # Add these classes only if vtkTypeUInt64Array exists
diff -ru VTK/IO/CMakeLists.txt VTK/IO/CMakeLists.txt
--- VTK/IO/CMakeLists.txt	2010-09-27 13:40:05.000000000 -0700
+++ VTK/IO/CMakeLists.txt	2012-02-23 20:47:45.000000000 -0800
@@ -18,11 +18,17 @@
 SET(KIT_PYTHON_LIBS vtkFilteringPythonD)
 SET(KIT_JAVA_LIBS vtkFilteringJava)
 SET(KIT_INTERFACE_LIBRARIES vtkFiltering)
+SET(KIT_INTERFACE_LIBRARIES_DEBUG vtkFiltering-gd)
 SET(KIT_LIBS vtkDICOMParser vtkNetCDF vtkNetCDF_cxx
   ${_VTK_METAIO_LIB} vtksqlite
   ${VTK_PNG_LIBRARIES} ${VTK_ZLIB_LIBRARIES} ${VTK_JPEG_LIBRARIES}
   ${VTK_TIFF_LIBRARIES} ${VTK_EXPAT_LIBRARIES} ${VTK_OGGTHEORA_LIBRARIES}
   ${KWSYS_NAMESPACE})
+SET(KIT_LIBS_DEBUG vtkDICOMParser-gd vtkNetCDF-gd vtkNetCDF_cxx-gd
+  ${_VTK_METAIO_LIB} vtksqlite-gd
+  ${VTK_PNG_LIBRARIES}-gd ${VTK_ZLIB_LIBRARIES}-gd ${VTK_JPEG_LIBRARIES}-gd
+  ${VTK_TIFF_LIBRARIES}-gd ${VTK_EXPAT_LIBRARIES}-gd ${VTK_OGGTHEORA_LIBRARIES}-gd
+  ${KWSYS_NAMESPACE}-gd)
 
 SET( Kit_SRCS
 vtkAbstractParticleWriter.cxx
@@ -248,6 +254,7 @@
   IF(VTK_USE_VIDEO_FOR_WINDOWS)
     SET(Kit_SRCS ${Kit_SRCS} vtkAVIWriter.cxx)
     SET(KIT_LIBS ${KIT_LIBS} vfw32)
+		SET(KIT_LIBS_DEBUG ${KIT_LIBS_DEBUG} vfw32)
   ENDIF(VTK_USE_VIDEO_FOR_WINDOWS)
 ENDIF(WIN32)
 
@@ -273,9 +280,15 @@
     ${FFMPEG_avformat_LIBRARY}
     ${FFMPEG_avcodec_LIBRARY}
     ${FFMPEG_avutil_LIBRARY})
+	SET(KIT_LIBS_DEBUG ${KIT_LIBS_DEBUG}
+    ${FFMPEG_avformat_LIBRARY}
+    ${FFMPEG_avcodec_LIBRARY}
+    ${FFMPEG_avutil_LIBRARY})
   IF (NOT VTK_FFMPEG_HAS_IMG_CONVERT)
     SET(KIT_LIBS ${KIT_LIBS}
       ${FFMPEG_swscale_LIBRARY})
+		SET(KIT_LIBS_DEBUG ${KIT_LIBS_DEBUG}
+      ${FFMPEG_swscale_LIBRARY})
   ENDIF (NOT VTK_FFMPEG_HAS_IMG_CONVERT)
 ENDIF (VTK_USE_FFMPEG_ENCODER)
 
@@ -290,6 +303,7 @@
   INCLUDE_DIRECTORIES(${vtkMPEG2Encode_INCLUDE_PATH})
   SET(Kit_SRCS ${Kit_SRCS} vtkMPEG2Writer.cxx)
   SET(KIT_LIBS ${KIT_LIBS} ${vtkMPEG2Encode_LIBRARIES})
+	SET(KIT_LIBS_DEBUG ${KIT_LIBS_DEBUG} ${vtkMPEG2Encode_LIBRARIES}-gd)
 ENDIF (VTK_USE_MPEG2_ENCODER)
 
 #-----------------------------------------------------------------------------
diff -ru VTK/Parallel/CMakeLists.txt VTK/Parallel/CMakeLists.txt
--- VTK/Parallel/CMakeLists.txt	2010-09-27 13:40:06.000000000 -0700
+++ VTK/Parallel/CMakeLists.txt	2012-02-23 20:47:46.000000000 -0800
@@ -4,19 +4,25 @@
 SET(KIT_PYTHON_LIBS vtkIOPythonD)
 SET(KIT_JAVA_LIBS vtkIOJava)
 SET(KIT_INTERFACE_LIBRARIES vtkIO)
+SET(KIT_INTERFACE_LIBRARIES_DEBUG vtkIO-gd)
 SET(KIT_LIBS VPIC Cosmo)
+SET(KIT_LIBS_DEBUG VPIC-gd Cosmo-gd)
 IF(VTK_USE_RENDERING)
   SET(KIT_TCL_LIBS vtkRenderingTCL ${KIT_TCL_LIBS})
   SET(KIT_PYTHON_LIBS vtkRenderingPythonD ${KIT_PYTHON_LIBS})
   SET(KIT_JAVA_LIBS vtkRenderingJava ${KIT_JAVA_LIBS})
   SET(KIT_INTERFACE_LIBRARIES vtkRendering ${KIT_INTERFACE_LIBRARIES})
+  SET(KIT_INTERFACE_LIBRARIES_DEBUG vtkRendering-gd ${KIT_INTERFACE_LIBRARIES_DEBUG})
   SET(KIT_LIBS ${KIT_LIBS} ${OPENGL_gl_LIBRARY})
+	SET(KIT_LIBS_DEBUG ${KIT_LIBS_DEBUG} ${OPENGL_gl_LIBRARY})
 ELSE(VTK_USE_RENDERING)
   SET(KIT_PYTHON_LIBS ${KIT_PYTHON_LIBS} vtkGraphicsPythonD)
   SET(KIT_LIBS ${KIT_LIBS} vtkGraphics vtkImaging)
+	SET(KIT_LIBS_DEBUG ${KIT_LIBS_DEBUG} vtkGraphics-gd vtkImaging-gd)
 ENDIF(VTK_USE_RENDERING)
 IF(VTK_HAS_EXODUS)
   SET(KIT_LIBS ${KIT_LIBS} vtkexoIIc)
+	SET(KIT_LIBS_DEBUG ${KIT_LIBS_DEBUG} vtkexoIIc-gd)
 ENDIF(VTK_HAS_EXODUS)
 
 SET ( Kit_SRCS
@@ -169,6 +175,7 @@
 
 IF (VTK_USE_PARALLEL_BGL)
   SET(KIT_LIBS ${KIT_LIBS} ${Boost_LIBRARIES})
+	SET(KIT_LIBS_DEBUG ${KIT_LIBS_DEBUG} ${Boost_LIBRARIES})
   SET ( Kit_SRCS
     ${Kit_SRCS}
     vtkPBGLBreadthFirstSearch.cxx
@@ -189,11 +196,14 @@
 IF (VTK_USE_MPI) 
   IF (MPI_LIBRARY) 
     SET(KIT_LIBS ${KIT_LIBS} "${MPI_LIBRARY}")   
+		SET(KIT_LIBS_DEBUG ${KIT_LIBS_DEBUG} "${MPI_LIBRARY}")   
   ELSE (MPI_LIBRARY)   
     MESSAGE("Could not find the required MPI libraries")
   ENDIF (MPI_LIBRARY)   
   IF (MPI_EXTRA_LIBRARY)   
     SET(KIT_LIBS ${KIT_LIBS} "${MPI_EXTRA_LIBRARY}")   
+		    SET(KIT_LIBS ${KIT_LIBS} "${MPI_EXTRA_LIBRARY}")   
+				SET(KIT_LIBS_DEBUG ${KIT_LIBS_DEBUG} "${MPI_EXTRA_LIBRARY}")   
   ENDIF (MPI_EXTRA_LIBRARY) 
 ENDIF (VTK_USE_MPI)
 
diff -ru VTK/Rendering/CMakeLists.txt VTK/Rendering/CMakeLists.txt
--- VTK/Rendering/CMakeLists.txt	2010-09-27 13:40:06.000000000 -0700
+++ VTK/Rendering/CMakeLists.txt	2012-02-23 20:47:46.000000000 -0800
@@ -9,9 +9,13 @@
 ENDIF (JAVA_AWT_LIBRARY)
 
 SET(KIT_INTERFACE_LIBRARIES vtkGraphics vtkImaging)
+SET(KIT_INTERFACE_LIBRARIES_DEBUG vtkGraphics-gd vtkImaging-gd)
 SET(KIT_LIBS vtkIO vtkftgl
   ${VTK_FREETYPE_LIBRARIES}
 )
+SET(KIT_LIBS_DEBUG vtkIO-gd vtkftgl-gd
+  ${VTK_FREETYPE_LIBRARIES_DDEBUG}
+)
 #INCLUDE(${VTK_CMAKE_DIR}/vtkTestGL.cmake)
 #INCLUDE(${VTK_CMAKE_DIR}/vtkTestGLX.cmake)
 
@@ -219,6 +223,7 @@
     )
   INCLUDE_DIRECTORIES(${CG_INCLUDE_PATH})
   SET (KIT_LIBS ${KIT_LIBS} ${CG_LIBRARY} ${CG_GL_LIBRARY})
+	SET (KIT_LIBS_DEBUG ${KIT_LIBS_DEBUG} ${CG_LIBRARY} ${CG_GL_LIBRARY})
 ENDIF(VTK_USE_CG_SHADERS)
 
 
@@ -376,6 +381,7 @@
    INCLUDE_DIRECTORIES(${VTK_TDX_INCLUDE_PATH})
    SET_SOURCE_FILES_PROPERTIES(vtkTDxDevice.cxx ABSTRACT)
    LIST(APPEND KIT_INTERFACE_LIBRARIES ${VTK_TDX_OBJECT_PATH})
+	 LIST(APPEND KIT_INTERFACE_LIBRARIES_DEBUG ${VTK_TDX_OBJECT_PATH})
   ENDIF()
 ENDIF()
 # -----------------------------------------------------------------------------
@@ -388,11 +394,13 @@
    find_package(3DconnexionClient)
    if(3DconnexionClient_FOUND)
      set(KIT_LIBS ${KIT_LIBS} ${3DconnexionClient_LIBRARY})
+     set(KIT_LIBS_DEBUG ${KIT_LIBS_DEBUG} ${3DconnexionClient_LIBRARY})
      include_directories(${3DconnexionClient_INCLUDE_DIR})
    endif()
    SET(Kit_SRCS ${Kit_SRCS} vtkTDxDevice.cxx vtkTDxMacDevice.cxx)
    SET_SOURCE_FILES_PROPERTIES(vtkTDxDevice.cxx ABSTRACT)
    LIST(APPEND KIT_INTERFACE_LIBRARIES ${VTK_TDX_OBJECT_PATH})
+   LIST(APPEND KIT_INTERFACE_LIBRARIES_DEBUG ${VTK_TDX_OBJECT_PATH})
   ENDIF()
 ENDIF()
 # -----------------------------------------------------------------------------
@@ -428,8 +436,10 @@
   # and executables that depend on vtkRendering to also directly link to Qt libraries.
   IF (APPLE)
     LIST (APPEND KIT_INTERFACE_LIBRARIES ${QT_LIBRARIES})
+		LIST (APPEND KIT_INTERFACE_LIBRARIES_DEBUG ${QT_LIBRARIES})
   ELSE (APPLE)
     LIST (APPEND KIT_LIBS ${QT_LIBRARIES})
+		LIST (APPEND KIT_LIBS_DEBUG ${QT_LIBRARIES})
   ENDIF (APPLE)
 
 ENDIF(VTK_USE_QT)
@@ -580,8 +590,10 @@
 IF(VTK_USE_GL2PS)
   IF(VTK_USE_SYSTEM_GL2PS)
     SET(KIT_LIBS ${KIT_LIBS} ${GL2PS_LIBRARIES} ${VTK_ZLIB_LIBRARIES} ${VTK_PNG_LIBRARIES})
+		SET(KIT_LIBS_DEBUG ${KIT_LIBS_DEBUG} ${GL2PS_LIBRARIES} vtkzlib-gd vtkpng-gd)
   ELSE(VTK_USE_SYSTEM_GL2PS)
     SET(KIT_LIBS ${KIT_LIBS} ${VTK_ZLIB_LIBRARIES} ${VTK_PNG_LIBRARIES})
+    SET(KIT_LIBS_DEBUG ${KIT_LIBS_DEBUG} vtkzlib-gd vtkpng-gd)
   ENDIF(VTK_USE_SYSTEM_GL2PS)
 ENDIF(VTK_USE_GL2PS)
 
@@ -590,12 +602,14 @@
     IF (OPENGL_gl_LIBRARY)
       SET(KitOpenGL_SRCS ${KitOpenGL_SRCS} vtkXOpenGLRenderWindow.cxx)
       SET(KIT_LIBS ${KIT_LIBS} ${OPENGL_gl_LIBRARY} )
+      SET(KIT_LIBS_DEBUG ${KIT_LIBS_DEBUG} ${OPENGL_gl_LIBRARY} )
     ENDIF (OPENGL_gl_LIBRARY)    
   ELSE(VTK_USE_X)
     SET(KitOpenGL_SRCS ${KitOpenGL_SRCS}
                  vtkWin32OpenGLRenderWindow.cxx
                  vtkWin32RenderWindowInteractor.cxx)
     SET (KIT_LIBS ${KIT_LIBS} ${OPENGL_gl_LIBRARY} )
+		SET (KIT_LIBS_DEBUG ${KIT_LIBS_DEBUG} ${OPENGL_gl_LIBRARY} )
   ENDIF(VTK_USE_X)
 ELSE (WIN32)
   IF (APPLE)
@@ -618,18 +632,21 @@
 
       SET_SOURCE_FILES_PROPERTIES(vtkCocoaGLView WRAP_EXCLUDE)
       SET (KIT_LIBS ${KIT_LIBS} ${OPENGL_gl_LIBRARY} "-framework Cocoa")
+      SET (KIT_LIBS_DEBUG ${KIT_LIBS_DEBUG} ${OPENGL_gl_LIBRARY} "-framework Cocoa")
       # This is to resolve a linker error that happens on older (?) OS X builds.
       # If an executable links against vtkRendering, it would get the following
       # error unless we add the objc to its link interface:
       # ld: /usr/lib/crt1.o illegal reference to symbol: __objcInit defined in indirectly
       # referenced dynamic library /usr/lib/libobjc.A.dylib
       LIST(APPEND KIT_INTERFACE_LIBRARIES objc)
+      LIST(APPEND KIT_INTERFACE_LIBRARIES_DEBUG objc)
     ENDIF(VTK_USE_COCOA)
     IF(VTK_USE_CARBON)
       SET(KitOpenGL_SRCS ${KitOpenGL_SRCS}
                    vtkCarbonRenderWindowInteractor.cxx
                    vtkCarbonRenderWindow.cxx)
       SET (KIT_LIBS ${KIT_LIBS} ${OPENGL_gl_LIBRARY} ${OPENGL_glu_LIBRARY} "-framework Carbon")
+			SET (KIT_LIBS_DEBUG ${KIT_LIBS_DEBUG} ${OPENGL_gl_LIBRARY} ${OPENGL_glu_LIBRARY} "-framework Carbon")
     ENDIF(VTK_USE_CARBON)
     IF(VTK_USE_X  AND  OPENGL_gl_LIBRARY)
       SET(KitOpenGL_SRCS ${KitOpenGL_SRCS} vtkXOpenGLRenderWindow.cxx)
@@ -638,10 +655,12 @@
     IF (VTK_USE_X AND OPENGL_gl_LIBRARY)
       SET(KitOpenGL_SRCS ${KitOpenGL_SRCS} vtkXOpenGLRenderWindow.cxx)
       SET (KIT_LIBS ${KIT_LIBS} ${OPENGL_gl_LIBRARY} )
+			SET (KIT_LIBS_DEBUG ${KIT_LIBS_DEBUG} ${OPENGL_gl_LIBRARY} )
     ELSE (VTK_USE_X AND OPENGL_gl_LIBRARY)
       IF (VTK_OPENGL_HAS_OSMESA  AND  OSMESA_LIBRARY)
         SET(KitOpenGL_SRCS ${KitOpenGL_SRCS} vtkOSOpenGLRenderWindow.cxx)
         SET (KIT_LIBS ${KIT_LIBS} ${OSMESA_LIBRARY} )
+				SET (KIT_LIBS_DEBUG ${KIT_LIBS_DEBUG} ${OSMESA_LIBRARY} )
       ENDIF (VTK_OPENGL_HAS_OSMESA  AND  OSMESA_LIBRARY)
     ENDIF (VTK_USE_X AND OPENGL_gl_LIBRARY)
   ENDIF (APPLE)
@@ -649,6 +668,7 @@
 
 IF(VTK_USE_CG_SHADERS)
   SET(KIT_LIBS ${KIT_LIBS} ${OPENGL_glu_LIBRARY})
+	SET(KIT_LIBS_DEBUG ${KIT_LIBS_DEBUG} ${OPENGL_glu_LIBRARY})
 ENDIF(VTK_USE_CG_SHADERS)
 
 if(DirectX_FOUND)
@@ -657,11 +677,13 @@
  # wbemuuid.lib is *NOT A FILE* (you cannot find it), it is probably an
  # embedded library added by the linker itself.
  set(KIT_LIBS ${KIT_LIBS} ${DirectX_LIBRARY} wbemuuid)
+ set(KIT_LIBS_DEBUG ${KIT_LIBS_DEBUG} ${DirectX_LIBRARY} wbemuuid)
  include_directories(${DirectX_INCLUDE_DIR})
 endif(DirectX_FOUND)
 
 if(VTK_USE_CORE_GRAPHICS)
  set(KIT_LIBS ${KIT_LIBS} ${ApplicationServices_LIBRARY} ${IOKit_LIBRARY})
+ set(KIT_LIBS_DEBUG ${KIT_LIBS_DEBUG} ${ApplicationServices_LIBRARY} ${IOKit_LIBRARY})
  include_directories(${ApplicationServices_INCLUDE_DIR} ${IOKit_INCLUDE_DIR})
 endif(VTK_USE_CORE_GRAPHICS)
 
@@ -669,6 +691,7 @@
  # NVCONTROL depends on X11. We solve that by making KIT_LIBS depending on
  # vtkRendering, as vtkRendering depends on X11 on Unix.
  set(KIT_LIBS ${KIT_LIBS} ${NVCtrlLib_LIBRARY} vtkRendering)
+ set(KIT_LIBS_DEBUG ${KIT_LIBS_DEBUG} ${NVCtrlLib_LIBRARY} vtkRendering-gd)
  include_directories(${NVCtrlLib_INCLUDE_DIR})
 endif(VTK_USE_NVCONTROL)
 
@@ -700,13 +723,16 @@
     )
     SET_SOURCE_FILES_PROPERTIES(vtkMesaRenderWindow ABSTRACT)
     SET (KIT_LIBS ${KIT_LIBS} ${MANGLED_MESA_LIBRARY})
+		SET (KIT_LIBS_DEBUG ${KIT_LIBS_DEBUG} ${MANGLED_MESA_LIBRARY})
     IF(MANGLED_OSMESA_LIBRARY)
       SET( KIT_LIBS ${KIT_LIBS} ${MANGLED_OSMESA_LIBRARY} )
+			SET( KIT_LIBS_DEBUG ${KIT_LIBS_DEBUG} ${MANGLED_OSMESA_LIBRARY} )
     ENDIF(MANGLED_OSMESA_LIBRARY)
   ENDIF (MANGLED_MESA_LIBRARY)
 ELSE (VTK_USE_MANGLED_MESA)
   IF(VTK_OPENGL_HAS_OSMESA  AND  OSMESA_LIBRARY)
     SET(KIT_LIBS ${KIT_LIBS} ${OSMESA_LIBRARY})
+		SET(KIT_LIBS_DEBUG ${KIT_LIBS_DEBUG} ${OSMESA_LIBRARY})
   ENDIF(VTK_OPENGL_HAS_OSMESA  AND  OSMESA_LIBRARY)
 ENDIF(VTK_USE_MANGLED_MESA)
 SET(Kit_SRCS ${Kit_SRCS} ${KitOpenGL_SRCS})
@@ -807,6 +833,9 @@
 
   ADD_LIBRARY(vtkRenderingPythonTkWidgets SHARED
               ${RenderingPythonTkWidgets_SRCS})
+  ADD_LIBRARY(vtkRenderingPythonTkWidgets-gd SHARED
+              ${RenderingPythonTkWidgets_SRCS})
+	SET_TARGET_PROPERTIES(vtkRenderingPythonTkWidgets-gd PROPERTIES COMPILE_FLAGS ${CMAKE_CXX_FLAGS_DEBUG})
   SET(VTK_PYTHONTK_LIBRARIES ${VTK_TK_LIBRARIES})
   IF(VTK_USE_X)
     SET(VTK_PYTHONTK_LIBRARIES ${VTK_PYTHONTK_LIBRARIES} ${X11_LIBRARIES})
@@ -818,11 +847,16 @@
   TARGET_LINK_LIBRARIES (vtkRenderingPythonTkWidgets
                         vtk${KIT}
                         ${VTK_PYTHONTK_LIBRARIES})
+  TARGET_LINK_LIBRARIES (vtkRenderingPythonTkWidgets-gd
+                        vtk${KIT}-gd
+                        ${VTK_PYTHONTK_LIBRARIES})
 
   # Apply user-defined properties to the library target.
   IF(VTK_LIBRARY_PROPERTIES)
     SET_TARGET_PROPERTIES(vtkRenderingPythonTkWidgets PROPERTIES
       ${VTK_LIBRARY_PROPERTIES})
+    SET_TARGET_PROPERTIES(vtkRenderingPythonTkWidgets-gd PROPERTIES
+      ${VTK_LIBRARY_PROPERTIES})
   ENDIF(VTK_LIBRARY_PROPERTIES)
 
   IF(NOT VTK_INSTALL_NO_LIBRARIES)
@@ -830,6 +864,10 @@
       RUNTIME DESTINATION ${VTK_INSTALL_BIN_DIR_CM24} COMPONENT RuntimeLibraries
       LIBRARY DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT RuntimeLibraries
       ARCHIVE DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT Development)
+    INSTALL(TARGETS vtkRenderingPythonTkWidgets-gd
+      RUNTIME DESTINATION ${VTK_INSTALL_BIN_DIR_CM24} COMPONENT RuntimeLibraries
+      LIBRARY DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT RuntimeLibraries
+      ARCHIVE DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT Development)
   ENDIF(NOT VTK_INSTALL_NO_LIBRARIES)
 ENDIF (VTK_WRAP_PYTHON AND VTK_USE_TK AND TK_FOUND)
 
@@ -844,8 +882,10 @@
   MARK_AS_ADVANCED(X11_Xt_LIB)
   IF (X11_Xt_LIB)
     TARGET_LINK_LIBRARIES(vtk${KIT} ${X11_Xt_LIB})
+    TARGET_LINK_LIBRARIES(vtk${KIT}-gd ${X11_Xt_LIB})
   ENDIF (X11_Xt_LIB)
   TARGET_LINK_LIBRARIES(vtk${KIT} ${X11_LIBRARIES})
+  TARGET_LINK_LIBRARIES(vtk${KIT}-gd ${X11_LIBRARIES})
   IF (VTK_USE_TK)
     # Definitely needed in case we are linking against a Tcl/Tk 8.5 static,
     # which by default seems to be missing Xss and Xft. It can not just be
diff -ru VTK/TextAnalysis/CMakeLists.txt VTK/TextAnalysis/CMakeLists.txt
--- VTK/TextAnalysis/CMakeLists.txt	2010-09-27 13:40:07.000000000 -0700
+++ VTK/TextAnalysis/CMakeLists.txt	2012-02-23 20:47:47.000000000 -0800
@@ -34,6 +34,11 @@
   vtkFiltering
   )
 
+SET(KIT_LIBS_DEBUG
+  vtkIO-gd
+  vtkFiltering-gd
+  )
+
 SET( Kit_SRCS
 vtkAssignMimeType.cxx
 vtkConcatenateArray.cxx
@@ -93,8 +98,10 @@
   # and executables that depend on vtkRendering to also directly link to Qt libraries.
   IF (APPLE)
     LIST (APPEND KIT_INTERFACE_LIBRARIES ${QT_LIBRARIES})
+    LIST (APPEND KIT_INTERFACE_LIBRARIES_DEBUG ${QT_LIBRARIES})
   ELSE (APPLE)
     LIST (APPEND KIT_LIBS ${QT_LIBRARIES})
+    LIST (APPEND KIT_LIBS_DEBUG ${QT_LIBRARIES})
   ENDIF (APPLE)
 ENDIF(VTK_USE_QT)
 
diff -ru VTK/Utilities/Cosmo/CMakeLists.txt VTK/Utilities/Cosmo/CMakeLists.txt
--- VTK/Utilities/Cosmo/CMakeLists.txt	2010-09-27 13:40:07.000000000 -0700
+++ VTK/Utilities/Cosmo/CMakeLists.txt	2012-02-23 20:47:47.000000000 -0800
@@ -33,13 +33,22 @@
 INCLUDE_DIRECTORIES (${Cosmo_SOURCE_DIR} ${Cosmo_BINARY_DIR})
 
 ADD_LIBRARY(Cosmo ${Cosmo_SOURCES})
+
 TARGET_LINK_LIBRARIES(Cosmo vtksys vtkCommon ${Cosmo_LIBS})
 
+ADD_LIBRARY(Cosmo-gd ${Cosmo_SOURCES})
+SET_TARGET_PROPERTIES(Cosmo-gd PROPERTIES COMPILE_FLAGS ${CMAKE_CXX_FLAGS_DEBUG})
+TARGET_LINK_LIBRARIES(Cosmo-gd vtksys-gd vtkCommon-gd ${Cosmo_LIBS})
+
 IF(NOT VTK_INSTALL_NO_LIBRARIES)
   INSTALL(TARGETS Cosmo
     RUNTIME DESTINATION ${VTK_INSTALL_BIN_DIR_CM24} COMPONENT RuntimeLibraries
     LIBRARY DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT RuntimeLibraries
     ARCHIVE DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT Development)
+  INSTALL(TARGETS Cosmo-gd
+    RUNTIME DESTINATION ${VTK_INSTALL_BIN_DIR_CM24} COMPONENT RuntimeLibraries
+    LIBRARY DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT RuntimeLibraries
+    ARCHIVE DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT Development)
 ENDIF(NOT VTK_INSTALL_NO_LIBRARIES)
 
 IF(NOT VTK_INSTALL_NO_DEVELOPMENT)
diff -ru VTK/Utilities/DICOMParser/CMakeLists.txt VTK/Utilities/DICOMParser/CMakeLists.txt
--- VTK/Utilities/DICOMParser/CMakeLists.txt	2010-09-27 13:40:07.000000000 -0700
+++ VTK/Utilities/DICOMParser/CMakeLists.txt	2012-02-23 20:47:47.000000000 -0800
@@ -27,10 +27,13 @@
                ${DICOMParser_BINARY_DIR}/DICOMCMakeConfig.h)
 
 ADD_LIBRARY(vtkDICOMParser DICOMFile.cxx DICOMParser.cxx DICOMAppHelper.cxx)
+ADD_LIBRARY(vtkDICOMParser-gd DICOMFile.cxx DICOMParser.cxx DICOMAppHelper.cxx)
+SET_TARGET_PROPERTIES(vtkDICOMParser-gd PROPERTIES COMPILE_FLAGS ${CMAKE_CXX_FLAGS_DEBUG})
 
 # Apply user-defined properties to the library target.
 IF(VTK_LIBRARY_PROPERTIES)
   SET_TARGET_PROPERTIES(vtkDICOMParser PROPERTIES ${VTK_LIBRARY_PROPERTIES})
+  SET_TARGET_PROPERTIES(vtkDICOMParser-gd PROPERTIES ${VTK_LIBRARY_PROPERTIES})
 ENDIF(VTK_LIBRARY_PROPERTIES)
 
 IF(NOT VTK_INSTALL_NO_LIBRARIES)
@@ -38,6 +41,10 @@
     RUNTIME DESTINATION ${VTK_INSTALL_BIN_DIR_CM24} COMPONENT RuntimeLibraries
     LIBRARY DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT RuntimeLibraries
     ARCHIVE DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT Development)
+  INSTALL(TARGETS vtkDICOMParser-gd
+    RUNTIME DESTINATION ${VTK_INSTALL_BIN_DIR_CM24} COMPONENT RuntimeLibraries
+    LIBRARY DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT RuntimeLibraries
+    ARCHIVE DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT Development)
 ENDIF(NOT VTK_INSTALL_NO_LIBRARIES)
 IF(NOT VTK_INSTALL_NO_DEVELOPMENT)
   FILE(GLOB __install_files.h "${DICOMParser_SOURCE_DIR}/*.h")
diff -ru VTK/Utilities/ftgl/CMakeLists.txt VTK/Utilities/ftgl/CMakeLists.txt
--- VTK/Utilities/ftgl/CMakeLists.txt	2010-09-27 13:40:08.000000000 -0700
+++ VTK/Utilities/ftgl/CMakeLists.txt	2012-02-23 20:47:47.000000000 -0800
@@ -146,10 +146,14 @@
 #
 ADD_LIBRARY (vtkftgl ${FTGL_SRCS})
 TARGET_LINK_LIBRARIES(vtkftgl ${VTKFTGL_LIBS})
+ADD_LIBRARY (vtkftgl-gd ${FTGL_SRCS})
+TARGET_LINK_LIBRARIES(vtkftgl-gd ${VTKFTGL_LIBS})
+SET_TARGET_PROPERTIES(vtkftgl-gd PROPERTIES COMPILE_FLAGS ${CMAKE_CXX_FLAGS_DEBUG})
 
 # Apply user-defined properties to the library target.
 IF(VTK_LIBRARY_PROPERTIES)
   SET_TARGET_PROPERTIES(vtkftgl PROPERTIES ${VTK_LIBRARY_PROPERTIES})
+  SET_TARGET_PROPERTIES(vtkftgl-gd PROPERTIES ${VTK_LIBRARY_PROPERTIES})
 ENDIF(VTK_LIBRARY_PROPERTIES)
 
 IF(NOT VTK_INSTALL_NO_LIBRARIES)
@@ -157,6 +161,10 @@
     RUNTIME DESTINATION ${VTK_INSTALL_BIN_DIR_CM24} COMPONENT RuntimeLibraries
     LIBRARY DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT RuntimeLibraries
     ARCHIVE DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT Development)
+  INSTALL(TARGETS vtkftgl-gd
+    RUNTIME DESTINATION ${VTK_INSTALL_BIN_DIR_CM24} COMPONENT RuntimeLibraries
+    LIBRARY DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT RuntimeLibraries
+    ARCHIVE DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT Development)
 ENDIF(NOT VTK_INSTALL_NO_LIBRARIES)
 
 #
diff -ru VTK/Utilities/kwsys/CMakeLists.txt VTK/Utilities/kwsys/CMakeLists.txt
--- VTK/Utilities/kwsys/CMakeLists.txt	2010-09-27 13:40:08.000000000 -0700
+++ VTK/Utilities/kwsys/CMakeLists.txt	2012-02-23 20:47:47.000000000 -0800
@@ -858,7 +858,11 @@
 IF(KWSYS_C_SRCS OR KWSYS_CXX_SRCS)
   ADD_LIBRARY(${KWSYS_NAMESPACE} ${KWSYS_LIBRARY_TYPE}
     ${KWSYS_C_SRCS} ${KWSYS_CXX_SRCS})
+  ADD_LIBRARY(${KWSYS_NAMESPACE}-gd ${KWSYS_LIBRARY_TYPE}
+    ${KWSYS_C_SRCS} ${KWSYS_CXX_SRCS})
+	SET_TARGET_PROPERTIES(${KWSYS_NAMESPACE}-gd PROPERTIES COMPILE_FLAGS ${CMAKE_CXX_FLAGS_DEBUG})
   KWSYS_SET_PROPERTY(TARGET ${KWSYS_NAMESPACE} PROPERTY LABELS ${KWSYS_LABELS_LIB})
+  KWSYS_SET_PROPERTY(TARGET ${KWSYS_NAMESPACE}-gd PROPERTY LABELS ${KWSYS_LABELS_LIB})
   IF(KWSYS_USE_DynamicLoader)
     IF(UNIX)
       TARGET_LINK_LIBRARIES(${KWSYS_NAMESPACE} ${CMAKE_DL_LIBS})
@@ -879,6 +883,7 @@
   # Create an install target for the library.
   IF(KWSYS_INSTALL_LIBRARY_RULE)
     INSTALL(TARGETS ${KWSYS_NAMESPACE} ${KWSYS_INSTALL_LIBRARY_RULE})
+    INSTALL(TARGETS ${KWSYS_NAMESPACE}-gd ${KWSYS_INSTALL_LIBRARY_RULE})
   ENDIF(KWSYS_INSTALL_LIBRARY_RULE)
   IF(KWSYS_LIBRARY_INSTALL_DIR)
     INSTALL_TARGETS(${KWSYS_LIBRARY_INSTALL_DIR} ${KWSYS_NAMESPACE})
@@ -888,21 +893,29 @@
 # Add a C-only library if requested.
 IF(KWSYS_ENABLE_C AND KWSYS_C_SRCS)
   ADD_LIBRARY(${KWSYS_NAMESPACE}_c ${KWSYS_LIBRARY_TYPE} ${KWSYS_C_SRCS})
+  ADD_LIBRARY(${KWSYS_NAMESPACE}_c-gd ${KWSYS_LIBRARY_TYPE} ${KWSYS_C_SRCS})
+	SET_TARGET_PROPERTIES(${KWSYS_NAMESPACE}_c-gd PROPERTIES COMPILE_FLAGS ${CMAKE_CXX_FLAGS_DEBUG})
   KWSYS_SET_PROPERTY(TARGET ${KWSYS_NAMESPACE}_c PROPERTY LABELS ${KWSYS_LABELS_LIB})
+  KWSYS_SET_PROPERTY(TARGET ${KWSYS_NAMESPACE}_c-gd PROPERTY LABELS ${KWSYS_LABELS_LIB})
 
   # Apply user-defined target properties to the library.
   IF(KWSYS_PROPERTIES_C)
     SET_TARGET_PROPERTIES(${KWSYS_NAMESPACE} PROPERTIES
       ${KWSYS_PROPERTIES_C}
       )
+    SET_TARGET_PROPERTIES(${KWSYS_NAMESPACE}-gd PROPERTIES
+      ${KWSYS_PROPERTIES_C}
+      )
   ENDIF(KWSYS_PROPERTIES_C)
 
   # Create an install target for the library.
   IF(KWSYS_INSTALL_LIBRARY_RULE)
     INSTALL(TARGETS ${KWSYS_NAMESPACE}_c ${KWSYS_INSTALL_LIBRARY_RULE})
+    INSTALL(TARGETS ${KWSYS_NAMESPACE}_c-gd ${KWSYS_INSTALL_LIBRARY_RULE})
   ENDIF(KWSYS_INSTALL_LIBRARY_RULE)
   IF(KWSYS_LIBRARY_INSTALL_DIR)
     INSTALL_TARGETS(${KWSYS_LIBRARY_INSTALL_DIR} ${KWSYS_NAMESPACE}_c)
+    INSTALL_TARGETS(${KWSYS_LIBRARY_INSTALL_DIR} ${KWSYS_NAMESPACE}_c-gd)
   ENDIF(KWSYS_LIBRARY_INSTALL_DIR)
 ENDIF(KWSYS_ENABLE_C AND KWSYS_C_SRCS)
 
diff -ru VTK/Utilities/mrmpi/mpistubs/CMakeLists.txt VTK/Utilities/mrmpi/mpistubs/CMakeLists.txt
--- VTK/Utilities/mrmpi/mpistubs/CMakeLists.txt	2010-09-27 13:40:08.000000000 -0700
+++ VTK/Utilities/mrmpi/mpistubs/CMakeLists.txt	2012-02-23 20:47:48.000000000 -0800
@@ -2,8 +2,12 @@
 
 ADD_LIBRARY (mpistubs ${mpistubs_SOURCES})
 TARGET_LINK_LIBRARIES (mpistubs vtksys)
+ADD_LIBRARY (mpistubs-gd ${mpistubs_SOURCES})
+TARGET_LINK_LIBRARIES (mpistubs-gd vtksys)
+SET_TARGET_PROPERTIES (mpistubs-gd PROPERTIES COMPILE_FLAGS ${CMAKE_CXX_FLAGS_DEBUG})
 IF(VTK_LIBRARY_PROPERTIES)
   SET_TARGET_PROPERTIES(mpistubs PROPERTIES ${VTK_LIBRARY_PROPERTIES})
+  SET_TARGET_PROPERTIES(mpistubs-gd PROPERTIES ${VTK_LIBRARY_PROPERTIES})
 ENDIF(VTK_LIBRARY_PROPERTIES)
 
 IF(NOT VTK_INSTALL_NO_LIBRARIES)
@@ -11,6 +15,10 @@
     RUNTIME DESTINATION ${VTK_INSTALL_BIN_DIR_CM24} COMPONENT RuntimeLibraries
     LIBRARY DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT RuntimeLibraries
     ARCHIVE DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT Development)
+  INSTALL(TARGETS mpistubs-gd
+    RUNTIME DESTINATION ${VTK_INSTALL_BIN_DIR_CM24} COMPONENT RuntimeLibraries
+    LIBRARY DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT RuntimeLibraries
+    ARCHIVE DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT Development)
 ENDIF(NOT VTK_INSTALL_NO_LIBRARIES)
 
 IF(NOT VTK_INSTALL_NO_DEVELOPMENT)
diff -ru VTK/Utilities/mrmpi/src/CMakeLists.txt VTK/Utilities/mrmpi/src/CMakeLists.txt
--- VTK/Utilities/mrmpi/src/CMakeLists.txt	2010-09-27 13:40:08.000000000 -0700
+++ VTK/Utilities/mrmpi/src/CMakeLists.txt	2012-02-23 20:47:48.000000000 -0800
@@ -8,9 +8,13 @@
         error.cpp)
 
 ADD_LIBRARY (MapReduceMPI ${mrmpi_SOURCES})
+ADD_LIBRARY (MapReduceMPI-gd ${mrmpi_SOURCES})
 TARGET_LINK_LIBRARIES (MapReduceMPI ${mrmpi_LIBS})
+TARGET_LINK_LIBRARIES (MapReduceMPI-gd ${mrmpi_LIBS})
+SET_TARGET_PROPERTIES (MapReduceMPI-gd PROPERTIES COMPILE_FLAGS ${CMAKE_CXX_FLAGS_DEBUG})
 IF(VTK_LIBRARY_PROPERTIES)
   SET_TARGET_PROPERTIES(MapReduceMPI PROPERTIES ${VTK_LIBRARY_PROPERTIES})
+  SET_TARGET_PROPERTIES(MapReduceMPI-gd PROPERTIES ${VTK_LIBRARY_PROPERTIES})
 ENDIF(VTK_LIBRARY_PROPERTIES)
 
 IF(NOT VTK_INSTALL_NO_LIBRARIES)
@@ -18,6 +22,10 @@
     RUNTIME DESTINATION ${VTK_INSTALL_BIN_DIR_CM24} COMPONENT RuntimeLibraries
     LIBRARY DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT RuntimeLibraries
     ARCHIVE DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT Development)
+  INSTALL(TARGETS MapReduceMPI-gd
+    RUNTIME DESTINATION ${VTK_INSTALL_BIN_DIR_CM24} COMPONENT RuntimeLibraries
+    LIBRARY DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT RuntimeLibraries
+    ARCHIVE DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT Development)
 ENDIF(NOT VTK_INSTALL_NO_LIBRARIES)
 
 IF(NOT VTK_INSTALL_NO_DEVELOPMENT)
diff -ru VTK/Utilities/verdict/CMakeLists.txt VTK/Utilities/verdict/CMakeLists.txt
--- VTK/Utilities/verdict/CMakeLists.txt	2010-09-27 13:40:08.000000000 -0700
+++ VTK/Utilities/verdict/CMakeLists.txt	2012-02-24 00:55:11.766866313 -0800
@@ -76,11 +76,16 @@
 endif ( NOT VERDICT_NO_LIBRARY_VERSION )
 
 add_library( ${verdict_LIBRARY} ${verdict_SRCS} )
+add_library( ${verdict_LIBRARY}-gd ${verdict_SRCS} )
+set_target_properties( ${verdict_LIBRARY}-gd PROPERTIES COMPILE_FLAGS ${CMAKE_CXX_FLAGS_DEBUG})
 # Apply user-defined properties to the library targets.
 if ( VERDICT_LIBRARY_PROPERTIES )
   set_target_properties( ${verdict_LIBRARY}
     PROPERTIES ${VERDICT_LIBRARY_PROPERTIES}
   )
+  set_target_properties( ${verdict_LIBRARY}-gd
+    PROPERTIES ${VERDICT_LIBRARY_PROPERTIES}
+  )
 endif ( VERDICT_LIBRARY_PROPERTIES )
 
 if ( VERDICT_ENABLE_TESTING )
@@ -91,10 +96,13 @@
   )
 
   add_executable( verdict_test ${verdict_TESTSRCS} )
-  target_link_libraries( verdict_test ${verdict_LIBRARY} )
+  add_executable( verdict_test-gd ${verdict_TESTSRCS} )
+  target_link_libraries( verdict_test ${verdict_LIBRARY}-gd )
 
   add_test( verdict_test verdict_test )
+  add_test( verdict_test-gd verdict_test-gd )
   set_tests_properties( verdict_test PROPERTIES FAIL_REGULAR_EXPRESSION "FAILED" )
+  set_tests_properties( verdict_test-gd PROPERTIES FAIL_REGULAR_EXPRESSION "FAILED" )
 endif ( VERDICT_ENABLE_TESTING )
 
 if (VTK_INSTALL_BIN_DIR_CM24)
@@ -163,6 +171,11 @@
       LIBRARY DESTINATION ${verdict_INSTALL_LIB_DIR} COMPONENT RuntimeLibraries # .so, mod.dll
       ARCHIVE DESTINATION ${verdict_INSTALL_LIB_DIR} COMPONENT Development      # .a, .lib
 )
+install(TARGETS ${verdict_LIBRARY}-gd
+      RUNTIME DESTINATION ${verdict_INSTALL_BIN_DIR} COMPONENT RuntimeLibraries # .exe, .dll
+      LIBRARY DESTINATION ${verdict_INSTALL_LIB_DIR} COMPONENT RuntimeLibraries # .so, mod.dll
+      ARCHIVE DESTINATION ${verdict_INSTALL_LIB_DIR} COMPONENT Development      # .a, .lib
+)
 
 #
 # Packing stuff
@@ -201,6 +214,11 @@
     set (CPACK_NSIS_URL_INFO_ABOUT "http:\\\\\\\\www.sandia.gov")
     set (CPACK_NSIS_CONTACT "cubit@sandia.gov")
     set (CPACK_NSIS_MODIFY_PATH ON)
+    if(CMAKE_SIZEOF_VOID_P EQUAL 8)
+      set(CPACK_NSIS_INSTALL_ROOT "$PROGRAMFILES64")
+    else()  
+      set(CPACK_NSIS_INSTALL_ROOT "$PROGRAMFILES32")
+    endif()
   else (WIN32 AND NOT UNIX)
     #set (CPACK_STRIP_FILES "bin/ccmake;bin/cmake;bin/cpack;bin/ctest")
     #set (CPACK_PACKAGE_EXECUTABLES "")
diff -ru VTK/Utilities/VPIC/CMakeLists.txt VTK/Utilities/VPIC/CMakeLists.txt
--- VTK/Utilities/VPIC/CMakeLists.txt	2010-09-27 13:40:08.000000000 -0700
+++ VTK/Utilities/VPIC/CMakeLists.txt	2012-02-23 20:47:49.000000000 -0800
@@ -24,10 +24,16 @@
 
 ADD_LIBRARY(VPIC ${VPIC_SOURCES})
 TARGET_LINK_LIBRARIES(VPIC vtksys )
+ADD_LIBRARY(VPIC-gd ${VPIC_SOURCES})
+SET_TARGET_PROPERTIES(VPIC-gd PROPERTIES COMPILE_FLAGS ${CMAKE_CXX_FLAGS_DEBUG})
+TARGET_LINK_LIBRARIES(VPIC-gd vtksys-gd )
+
 IF(VTK_USE_MPI)
 TARGET_LINK_LIBRARIES(VPIC ${MPI_LIBRARY})
+TARGET_LINK_LIBRARIES(VPIC-gd ${MPI_LIBRARY})
   IF(MPI_EXTRA_LIBRARY)
     TARGET_LINK_LIBRARIES(VPIC ${MPI_EXTRA_LIBRARY})
+    TARGET_LINK_LIBRARIES(VPIC-gd ${MPI_EXTRA_LIBRARY})
   ENDIF(MPI_EXTRA_LIBRARY)
 ENDIF(VTK_USE_MPI)
 
@@ -36,6 +42,10 @@
     RUNTIME DESTINATION ${VTK_INSTALL_BIN_DIR_CM24} COMPONENT RuntimeLibraries
     LIBRARY DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT RuntimeLibraries
     ARCHIVE DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT Development)
+  INSTALL(TARGETS VPIC-gd
+    RUNTIME DESTINATION ${VTK_INSTALL_BIN_DIR_CM24} COMPONENT RuntimeLibraries
+    LIBRARY DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT RuntimeLibraries
+    ARCHIVE DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT Development)
 ENDIF(NOT VTK_INSTALL_NO_LIBRARIES)
 
 IF(NOT VTK_INSTALL_NO_DEVELOPMENT)
diff -ru VTK/Utilities/vtkalglib/CMakeLists.txt VTK/Utilities/vtkalglib/CMakeLists.txt
--- VTK/Utilities/vtkalglib/CMakeLists.txt	2010-09-27 13:40:09.000000000 -0700
+++ VTK/Utilities/vtkalglib/CMakeLists.txt	2012-02-23 20:47:49.000000000 -0800
@@ -36,10 +36,15 @@
 add_library( vtkalglib
   ${ALGLIB_SRCS}
 )
+add_library( vtkalglib-gd
+  ${ALGLIB_SRCS}
+)
+set_target_properties(vtkalglib-gd PROPERTIES COMPILE_FLAGS ${CMAKE_CXX_FLAGS_DEBUG})
 
 # Apply user-defined properties to the library target.
 if ( VTK_LIBRARY_PROPERTIES )
   set_target_properties( vtkalglib PROPERTIES ${VTK_LIBRARY_PROPERTIES} )
+  set_target_properties( vtkalglib-gd PROPERTIES ${VTK_LIBRARY_PROPERTIES} )
 endif ( VTK_LIBRARY_PROPERTIES )
 
 #
@@ -71,6 +76,11 @@
     RUNTIME DESTINATION ${alglib_install_bin_dir} COMPONENT RuntimeLibraries
     LIBRARY DESTINATION ${alglib_install_lib_dir} COMPONENT RuntimeLibraries
     ARCHIVE DESTINATION ${alglib_install_lib_dir} COMPONENT Development
+  )
+  install( TARGETS vtkalglib-gd
+    RUNTIME DESTINATION ${alglib_install_bin_dir} COMPONENT RuntimeLibraries
+    LIBRARY DESTINATION ${alglib_install_lib_dir} COMPONENT RuntimeLibraries
+    ARCHIVE DESTINATION ${alglib_install_lib_dir} COMPONENT Development
   )
 endif ( NOT VTK_INSTALL_NO_LIBRARIES )
 
diff -ru VTK/Utilities/vtkexodus2/CMakeLists.txt VTK/Utilities/vtkexodus2/CMakeLists.txt
--- VTK/Utilities/vtkexodus2/CMakeLists.txt	2010-09-27 13:40:09.000000000 -0700
+++ VTK/Utilities/vtkexodus2/CMakeLists.txt	2012-02-23 20:47:49.000000000 -0800
@@ -214,10 +214,14 @@
 
 ADD_LIBRARY(vtkexoIIc ${cbind_SRCS})
 TARGET_LINK_LIBRARIES(vtkexoIIc vtkNetCDF)
+ADD_LIBRARY(vtkexoIIc-gd ${cbind_SRCS})
+TARGET_LINK_LIBRARIES(vtkexoIIc-gd vtkNetCDF-gd)
+SET_TARGET_PROPERTIES(vtkexoIIc-gd PROPERTIES COMPILE_FLAGS ${CMAKE_CXX_FLAGS_DEBUG})
 
 # Apply user-defined properties to the library target.
 IF(VTK_LIBRARY_PROPERTIES)
   SET_TARGET_PROPERTIES(vtkexoIIc PROPERTIES ${VTK_LIBRARY_PROPERTIES})
+  SET_TARGET_PROPERTIES(vtkexoIIc-gd PROPERTIES ${VTK_LIBRARY_PROPERTIES})
 ENDIF(VTK_LIBRARY_PROPERTIES)
 
 IF(NOT VTK_INSTALL_NO_LIBRARIES)
@@ -225,6 +229,10 @@
     RUNTIME DESTINATION ${VTK_INSTALL_BIN_DIR_CM24} COMPONENT RuntimeLibraries
     LIBRARY DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT RuntimeLibraries
     ARCHIVE DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT Development)
+  INSTALL(TARGETS vtkexoIIc-gd
+    RUNTIME DESTINATION ${VTK_INSTALL_BIN_DIR_CM24} COMPONENT RuntimeLibraries
+    LIBRARY DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT RuntimeLibraries
+    ARCHIVE DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT Development)
 ENDIF(NOT VTK_INSTALL_NO_LIBRARIES)
 IF(NOT VTK_INSTALL_NO_DEVELOPMENT)
 #  INSTALL_FILES(${VTK_INSTALL_INCLUDE_DIR}/vtkexodus2 .h
diff -ru VTK/Utilities/vtkexpat/CMakeLists.txt VTK/Utilities/vtkexpat/CMakeLists.txt
--- VTK/Utilities/vtkexpat/CMakeLists.txt	2010-09-27 13:40:09.000000000 -0700
+++ VTK/Utilities/vtkexpat/CMakeLists.txt	2012-02-23 20:47:49.000000000 -0800
@@ -23,10 +23,13 @@
                ${VTKEXPAT_BINARY_DIR}/.NoDartCoverage)
 
 ADD_LIBRARY(vtkexpat ${expat_SRCS})
+ADD_LIBRARY(vtkexpat-gd ${expat_SRCS})
+SET_TARGET_PROPERTIES(vtkexpat-gd PROPERTIES COMPILE_FLAGS ${CMAKE_CXX_FLAGS_DEBUG})
 
 # Apply user-defined properties to the library target.
 IF(VTK_LIBRARY_PROPERTIES)
   SET_TARGET_PROPERTIES(vtkexpat PROPERTIES ${VTK_LIBRARY_PROPERTIES})
+  SET_TARGET_PROPERTIES(vtkexpat-gd PROPERTIES ${VTK_LIBRARY_PROPERTIES})
 ENDIF(VTK_LIBRARY_PROPERTIES)
 
 IF(NOT VTK_INSTALL_NO_LIBRARIES)
@@ -34,6 +37,10 @@
     RUNTIME DESTINATION ${VTK_INSTALL_BIN_DIR_CM24} COMPONENT RuntimeLibraries
     LIBRARY DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT RuntimeLibraries
     ARCHIVE DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT Development)
+  INSTALL(TARGETS vtkexpat-gd
+    RUNTIME DESTINATION ${VTK_INSTALL_BIN_DIR_CM24} COMPONENT RuntimeLibraries
+    LIBRARY DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT RuntimeLibraries
+    ARCHIVE DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT Development)
 ENDIF(NOT VTK_INSTALL_NO_LIBRARIES)
 IF(NOT VTK_INSTALL_NO_DEVELOPMENT)
   INSTALL(FILES
diff -ru VTK/Utilities/vtkfreetype/CMakeLists.txt VTK/Utilities/vtkfreetype/CMakeLists.txt
--- VTK/Utilities/vtkfreetype/CMakeLists.txt	2010-09-27 13:40:09.000000000 -0700
+++ VTK/Utilities/vtkfreetype/CMakeLists.txt	2012-02-23 20:47:49.000000000 -0800
@@ -100,18 +100,21 @@
 # Define the library (and install it)
 #
 ADD_LIBRARY (vtkfreetype ${FREETYPE_SRCS})
-
+ADD_LIBRARY (vtkfreetype-gd ${FREETYPE_SRCS})
+SET_TARGET_PROPERTIES (vtkfreetype-gd PROPERTIES COMPILE_FLAGS ${CMAKE_CXX_FLAGS_DEBUG})
 #
 # If we want freetype's Mac font support we need to link against
 # a couple of Mac OS X frameworks.
 #
 IF (APPLE AND VTKFREETYPE_MAC_FONT_SUPPORT)
     TARGET_LINK_LIBRARIES (vtkfreetype "-framework ApplicationServices -framework CoreServices")
+    TARGET_LINK_LIBRARIES (vtkfreetype-gd "-framework ApplicationServices -framework CoreServices")
 ENDIF (APPLE AND VTKFREETYPE_MAC_FONT_SUPPORT)
 
 # Apply user-defined properties to the library target.
 IF(VTK_LIBRARY_PROPERTIES)
   SET_TARGET_PROPERTIES(vtkfreetype PROPERTIES ${VTK_LIBRARY_PROPERTIES})
+  SET_TARGET_PROPERTIES(vtkfreetype-gd PROPERTIES ${VTK_LIBRARY_PROPERTIES})
 ENDIF(VTK_LIBRARY_PROPERTIES)
 
 IF(NOT VTK_INSTALL_NO_LIBRARIES)
@@ -119,6 +122,10 @@
     RUNTIME DESTINATION ${VTK_INSTALL_BIN_DIR_CM24} COMPONENT RuntimeLibraries
     LIBRARY DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT RuntimeLibraries
     ARCHIVE DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT Development)
+  INSTALL(TARGETS vtkfreetype-gd
+    RUNTIME DESTINATION ${VTK_INSTALL_BIN_DIR_CM24} COMPONENT RuntimeLibraries
+    LIBRARY DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT RuntimeLibraries
+    ARCHIVE DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT Development)
 ENDIF(NOT VTK_INSTALL_NO_LIBRARIES)
 
 #
diff -ru VTK/Utilities/vtkjpeg/CMakeLists.txt VTK/Utilities/vtkjpeg/CMakeLists.txt
--- VTK/Utilities/vtkjpeg/CMakeLists.txt	2010-09-27 13:40:09.000000000 -0700
+++ VTK/Utilities/vtkjpeg/CMakeLists.txt	2012-02-23 20:47:50.000000000 -0800
@@ -46,10 +46,13 @@
                ${VTKJPEG_BINARY_DIR}/jpegDllConfig.h)
 
 ADD_LIBRARY(vtkjpeg ${systemdependent_SRCS} ${common_SRCS} ${compression_SRCS} ${decompression_SRCS})
+ADD_LIBRARY(vtkjpeg-gd ${systemdependent_SRCS} ${common_SRCS} ${compression_SRCS} ${decompression_SRCS})
+SET_TARGET_PROPERTIES(vtkjpeg-gd PROPERTIES COMPILE_FLAGS ${CMAKE_CXX_FLAGS_DEBUG})
 
 # Apply user-defined properties to the library target.
 IF(VTK_LIBRARY_PROPERTIES)
   SET_TARGET_PROPERTIES(vtkjpeg PROPERTIES ${VTK_LIBRARY_PROPERTIES})
+  SET_TARGET_PROPERTIES(vtkjpeg-gd PROPERTIES ${VTK_LIBRARY_PROPERTIES})
 ENDIF(VTK_LIBRARY_PROPERTIES)
 
 IF(NOT VTK_INSTALL_NO_LIBRARIES)
@@ -57,6 +60,10 @@
     RUNTIME DESTINATION ${VTK_INSTALL_BIN_DIR_CM24} COMPONENT RuntimeLibraries
     LIBRARY DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT RuntimeLibraries
     ARCHIVE DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT Development)
+  INSTALL(TARGETS vtkjpeg-gd
+    RUNTIME DESTINATION ${VTK_INSTALL_BIN_DIR_CM24} COMPONENT RuntimeLibraries
+    LIBRARY DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT RuntimeLibraries
+    ARCHIVE DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT Development)
 ENDIF(NOT VTK_INSTALL_NO_LIBRARIES)
 IF(NOT VTK_INSTALL_NO_DEVELOPMENT)
   INSTALL(FILES
diff -ru VTK/Utilities/vtklibproj4/CMakeLists.txt VTK/Utilities/vtklibproj4/CMakeLists.txt
--- VTK/Utilities/vtklibproj4/CMakeLists.txt	2010-09-27 13:40:09.000000000 -0700
+++ VTK/Utilities/vtklibproj4/CMakeLists.txt	2012-02-23 20:47:50.000000000 -0800
@@ -183,11 +183,18 @@
 add_library( vtkproj4
   ${LP4_SRCS}
 )
+add_library( vtkproj4-gd
+  ${LP4_SRCS}
+)
+set_target_properties( vtkproj4-gd PROPERTIES COMPILE_FLAGS ${CMAKE_CXX_FLAGS_DEBUG})
 
 if ( UNIX )
   target_link_libraries( vtkproj4
     m
   )
+  target_link_libraries( vtkproj4-gd
+    m
+  )
 endif ( UNIX )
 
 if ( PROJ_LIBRARY_PROPERTIES )
@@ -195,6 +202,10 @@
     PROPERTIES
       ${PROJ_LIBRARY_PROPERTIES}
   )
+  set_target_properties( vtkproj4-gd
+    PROPERTIES
+      ${PROJ_LIBRARY_PROPERTIES}
+  )
 endif ( PROJ_LIBRARY_PROPERTIES )
 
 add_executable( lproj
@@ -204,6 +215,14 @@
 target_link_libraries( lproj
   vtkproj4
 )
+add_executable( lproj-gd
+  lproj.c
+)
+
+target_link_libraries( lproj-gd
+  vtkproj4
+)
+set_target_properties(lproj-gd PROPERTIES COMPILE_FLAGS ${CMAKE_CXX_FLAGS_DEBUG})
 
 include_directories(
   ${libproj4_SOURCE_DIR}
@@ -215,6 +234,9 @@
   target_link_libraries( proj4
     ${GSL_LIBRARIES}
   )
+  target_link_libraries( proj4-gd
+    ${GSL_LIBRARIES}
+  )
 endif ( GSL_FOUND )
 
 configure_file(
@@ -235,6 +257,10 @@
     RUNTIME DESTINATION ${VTK_INSTALL_BIN_DIR_CM24} COMPONENT RuntimeLibraries
     LIBRARY DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT RuntimeLibraries
     ARCHIVE DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT Development)
+  INSTALL(TARGETS vtkproj4-gd lproj-gd
+    RUNTIME DESTINATION ${VTK_INSTALL_BIN_DIR_CM24} COMPONENT RuntimeLibraries
+    LIBRARY DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT RuntimeLibraries
+    ARCHIVE DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT Development)
 ENDIF(NOT VTK_INSTALL_NO_LIBRARIES)
 IF(NOT VTK_INSTALL_NO_DEVELOPMENT)
   INSTALL(FILES
diff -ru VTK/Utilities/vtklibxml2/CMakeLists.txt VTK/Utilities/vtklibxml2/CMakeLists.txt
--- VTK/Utilities/vtklibxml2/CMakeLists.txt	2010-09-27 13:40:10.000000000 -0700
+++ VTK/Utilities/vtklibxml2/CMakeLists.txt	2012-02-23 20:47:50.000000000 -0800
@@ -297,10 +297,14 @@
 ENDIF(NOT BUILD_SHARED_LIBS)
 ADD_LIBRARY(vtklibxml2 ${vtklibxml2_la_SOURCES})
 TARGET_LINK_LIBRARIES(vtklibxml2 ${LIBXML2_LIBS})
+ADD_LIBRARY(vtklibxml2-gd ${vtklibxml2_la_SOURCES})
+TARGET_LINK_LIBRARIES(vtklibxml2-gd ${LIBXML2_LIBS})
+SET_TARGET_PROPERTIES(vtklibxml2-gd PROPERTIES COMPILE_FLAGS ${CMAKE_CXX_FLAGS_DEBUG})
 
 # Apply user-defined properties to the library target.
 IF(VTK_LIBRARY_PROPERTIES)
   SET_TARGET_PROPERTIES(vtklibxml2 PROPERTIES ${VTK_LIBRARY_PROPERTIES})
+  SET_TARGET_PROPERTIES(vtklibxml2-gd PROPERTIES ${VTK_LIBRARY_PROPERTIES})
 ENDIF(VTK_LIBRARY_PROPERTIES)
 
 IF(NOT VTK_INSTALL_NO_LIBRARIES)
@@ -308,6 +312,10 @@
     RUNTIME DESTINATION ${VTK_INSTALL_BIN_DIR_CM24} COMPONENT RuntimeLibraries
     LIBRARY DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT RuntimeLibraries
     ARCHIVE DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT Development)
+  INSTALL(TARGETS vtklibxml2-gd
+    RUNTIME DESTINATION ${VTK_INSTALL_BIN_DIR_CM24} COMPONENT RuntimeLibraries
+    LIBRARY DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT RuntimeLibraries
+    ARCHIVE DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT Development)
 ENDIF(NOT VTK_INSTALL_NO_LIBRARIES)
 IF(NOT VTK_INSTALL_NO_DEVELOPMENT)
   INSTALL(FILES
diff -ru VTK/Utilities/vtkmetaio/CMakeLists.txt VTK/Utilities/vtkmetaio/CMakeLists.txt
--- VTK/Utilities/vtkmetaio/CMakeLists.txt	2010-09-27 13:40:10.000000000 -0700
+++ VTK/Utilities/vtkmetaio/CMakeLists.txt	2012-02-23 20:47:51.000000000 -0800
@@ -52,12 +52,48 @@
   metaArray.cxx
 )
 
+ADD_LIBRARY(${METAIO_NAMESPACE}-gd
+  metaTypes.h
+  metaUtils.cxx
+  metaArrow.cxx
+  metaBlob.cxx
+  metaCommand.h
+  metaCommand.cxx
+  metaContour.h
+  metaContour.cxx
+  metaDTITube.cxx
+  metaEllipse.cxx
+  metaGroup.cxx
+  metaGaussian.cxx
+  metaImage.cxx
+  metaImageUtils.cxx
+  metaLandmark.cxx
+  metaLine.cxx
+  metaMesh.cxx
+  metaMesh.h
+  metaObject.cxx
+  metaOutput.h
+  metaOutput.cxx
+  metaScene.cxx
+  metaSurface.cxx
+  metaTube.cxx
+  metaVesselTube.cxx
+  metaTransform.cxx
+  metaTubeGraph.cxx
+  metaForm.h
+  metaForm.cxx
+  metaArray.h
+  metaArray.cxx
+)
+SET_TARGET_PROPERTIES(${METAIO_NAMESPACE}-gd PROPERTIES COMPILE_FLAGS ${CMAKE_CXX_FLAGS_DEBUG})
+
 INCLUDE_REGULAR_EXPRESSION("^.*$")
 
 # Need nsl to resolve gethostbyname on SunOS-5.8
 # and socket also
 IF(CMAKE_SYSTEM MATCHES "SunOS.*")
   TARGET_LINK_LIBRARIES(${METAIO_NAMESPACE} socket nsl)
+  TARGET_LINK_LIBRARIES(${METAIO_NAMESPACE}-gd socket nsl)
 ENDIF(CMAKE_SYSTEM MATCHES "SunOS.*") 
 
 IF(METAIO_FOR_VTK)
@@ -66,33 +102,46 @@
   ENDIF(BUILD_SHARED_LIBS)
   TARGET_LINK_LIBRARIES(${METAIO_NAMESPACE} ${VTK_ZLIB_LIBRARIES}
                                             ${KWSYS_NAMESPACE})
+  TARGET_LINK_LIBRARIES(${METAIO_NAMESPACE}-gd ${VTK_ZLIB_LIBRARIES_DEBUG}
+                                            ${KWSYS_NAMESPACE}-gd)
   IF(VTK_LIBRARY_PROPERTIES)
     SET_TARGET_PROPERTIES(${METAIO_NAMESPACE} 
                           PROPERTIES ${VTK_LIBRARY_PROPERTIES})
+    SET_TARGET_PROPERTIES(${METAIO_NAMESPACE}-gd 
+                          PROPERTIES ${VTK_LIBRARY_PROPERTIES})
   ENDIF(VTK_LIBRARY_PROPERTIES)
 ELSE(METAIO_FOR_VTK)
   IF(METAIO_FOR_ITK)
     TARGET_LINK_LIBRARIES(${METAIO_NAMESPACE} ${ITK_ZLIB_LIBRARIES} 
                                               ${KWSYS_NAMESPACE})
+    TARGET_LINK_LIBRARIES(${METAIO_NAMESPACE}-gd ${VTK_ZLIB_LIBRARIES_DEBUG}
+                                                 ${KWSYS_NAMESPACE}-gd)
     IF(ITK_USE_LIBXML2)    
       TARGET_LINK_LIBRARIES(${METAIO_NAMESPACE} ${ITK_LIBXML2_LIBRARIES})
+      TARGET_LINK_LIBRARIES(${METAIO_NAMESPACE}-gd ${ITK_LIBXML2_LIBRARIES})
     ENDIF(ITK_USE_LIBXML2)   
 
     IF(ITK_LIBRARY_PROPERTIES)
       SET_TARGET_PROPERTIES(${METAIO_NAMESPACE} 
                             PROPERTIES ${ITK_LIBRARY_PROPERTIES})
+      SET_TARGET_PROPERTIES(${METAIO_NAMESPACE}-gd 
+                            PROPERTIES ${ITK_LIBRARY_PROPERTIES})
     ENDIF(ITK_LIBRARY_PROPERTIES)
   ELSE(METAIO_FOR_ITK)
     TARGET_LINK_LIBRARIES(${METAIO_NAMESPACE} ${ZLIB_LIBRARIES}
                                               ${KWSYS_NAMESPACE})
+    TARGET_LINK_LIBRARIES(${METAIO_NAMESPACE}-gd ${VTK_ZLIB_LIBRARIES}
+                                              ${KWSYS_NAMESPACE}-gd)
   ENDIF(METAIO_FOR_ITK)
 ENDIF(METAIO_FOR_VTK)
 
 IF(WIN32)
   IF(BORLAND)
     TARGET_LINK_LIBRARIES(${METAIO_NAMESPACE} import32)
+    TARGET_LINK_LIBRARIES(${METAIO_NAMESPACE}-gd import32)
   ELSE(BORLAND)
     TARGET_LINK_LIBRARIES(${METAIO_NAMESPACE} comctl32 wsock32)
+    TARGET_LINK_LIBRARIES(${METAIO_NAMESPACE}-gd comctl32 wsock32)
   ENDIF(BORLAND)
 ENDIF(WIN32)
 
@@ -109,6 +158,14 @@
             ARCHIVE DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} 
                     COMPONENT Development
             )
+    INSTALL(TARGETS ${METAIO_NAMESPACE}-gd
+            RUNTIME DESTINATION ${VTK_INSTALL_BIN_DIR_CM24} 
+                    COMPONENT RuntimeLibraries
+            LIBRARY DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} 
+                    COMPONENT RuntimeLibraries
+            ARCHIVE DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} 
+                    COMPONENT Development
+            )
   ENDIF(NOT VTK_INSTALL_NO_LIBRARIES)
   IF(NOT VTK_INSTALL_NO_DEVELOPMENT)
     FILE(GLOB __files "${CMAKE_CURRENT_SOURCE_DIR}/*.h")
@@ -127,6 +184,14 @@
               ARCHIVE DESTINATION ${ITK_INSTALL_LIB_DIR_CM24} 
                       COMPONENT Development
               )
+      INSTALL(TARGETS ${METAIO_NAMESPACE}-gd
+              RUNTIME DESTINATION ${ITK_INSTALL_BIN_DIR_CM24} 
+                      COMPONENT RuntimeLibraries
+              LIBRARY DESTINATION ${ITK_INSTALL_LIB_DIR_CM24} 
+                      COMPONENT RuntimeLibraries
+              ARCHIVE DESTINATION ${ITK_INSTALL_LIB_DIR_CM24} 
+                      COMPONENT Development
+              )
     ENDIF(NOT ITK_INSTALL_NO_LIBRARIES)
     IF(NOT ITK_INSTALL_NO_DEVELOPMENT)
       FILE(GLOB __files "${CMAKE_CURRENT_SOURCE_DIR}/*.h")
@@ -143,6 +208,14 @@
             RUNTIME DESTINATION ${INSTALL_BIN_DIR_CM24} 
                     COMPONENT RuntimeLibraries
             LIBRARY DESTINATION ${INSTALL_LIB_DIR_CM24} 
+                    COMPONENT RuntimeLibraries
+            ARCHIVE DESTINATION ${INSTALL_LIB_DIR_CM24} 
+                    COMPONENT Development
+            )
+    INSTALL(TARGETS ${METAIO_NAMESPACE}-gd
+            RUNTIME DESTINATION ${INSTALL_BIN_DIR_CM24} 
+                    COMPONENT RuntimeLibraries
+            LIBRARY DESTINATION ${INSTALL_LIB_DIR_CM24} 
                     COMPONENT RuntimeLibraries
             ARCHIVE DESTINATION ${INSTALL_LIB_DIR_CM24} 
                     COMPONENT Development
diff -ru VTK/Utilities/vtknetcdf/CMakeLists.txt VTK/Utilities/vtknetcdf/CMakeLists.txt
--- VTK/Utilities/vtknetcdf/CMakeLists.txt	2010-09-27 13:40:10.000000000 -0700
+++ VTK/Utilities/vtknetcdf/CMakeLists.txt	2012-02-23 20:47:51.000000000 -0800
@@ -288,9 +288,15 @@
 
 ADD_LIBRARY(vtkNetCDF
   ${netcdf_SRCS})
+ADD_LIBRARY(vtkNetCDF-gd
+  ${netcdf_SRCS})
+SET_TARGET_PROPERTIES(vtkNetCDF-gd PROPERTIES COMPILE_FLAGS ${CMAKE_CXX_FLAGS_DEBUG})
 
 ADD_LIBRARY(vtkNetCDF_cxx ${netcdfcxx_SRCS})
 TARGET_LINK_LIBRARIES(vtkNetCDF_cxx vtkNetCDF)
+ADD_LIBRARY(vtkNetCDF_cxx-gd ${netcdfcxx_SRCS})
+SET_TARGET_PROPERTIES(vtkNetCDF_cxx-gd PROPERTIES COMPILE_FLAGS ${CMAKE_CXX_FLAGS_DEBUG})
+TARGET_LINK_LIBRARIES(vtkNetCDF_cxx-gd vtkNetCDF-gd)
 
 # Take care of symbol export/import on Windows
 ADD_DEFINITIONS("-DNC_DLL_EXPORT")
@@ -298,6 +304,7 @@
 # Apply user-defined properties to the library target.
 IF(VTK_LIBRARY_PROPERTIES)
   SET_TARGET_PROPERTIES(vtkNetCDF PROPERTIES ${VTK_LIBRARY_PROPERTIES})
+  SET_TARGET_PROPERTIES(vtkNetCDF-gd PROPERTIES ${VTK_LIBRARY_PROPERTIES})
 ENDIF(VTK_LIBRARY_PROPERTIES)
 
 IF(NOT VTK_INSTALL_NO_LIBRARIES)
@@ -311,6 +318,10 @@
     RUNTIME DESTINATION ${VTK_INSTALL_BIN_DIR_CM24} COMPONENT RuntimeLibraries
     LIBRARY DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT RuntimeLibraries
     ARCHIVE DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT Development)
+  INSTALL(TARGETS vtkNetCDF-gd vtkNetCDF_cxx-gd
+    RUNTIME DESTINATION ${VTK_INSTALL_BIN_DIR_CM24} COMPONENT RuntimeLibraries
+    LIBRARY DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT RuntimeLibraries
+    ARCHIVE DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT Development)
 ENDIF(NOT VTK_INSTALL_NO_LIBRARIES)
 IF(NOT VTK_INSTALL_NO_DEVELOPMENT)
   INSTALL(FILES
diff -ru VTK/Utilities/vtkoggtheora/CMakeLists.txt VTK/Utilities/vtkoggtheora/CMakeLists.txt
--- VTK/Utilities/vtkoggtheora/CMakeLists.txt	2010-09-27 13:40:10.000000000 -0700
+++ VTK/Utilities/vtkoggtheora/CMakeLists.txt	2012-02-23 20:47:52.000000000 -0800
@@ -249,10 +249,12 @@
 ADD_DEFINITIONS(-DHAVE_CONFIG_H)
 
 ADD_LIBRARY(vtkoggtheora ${VTKOGGTHEORA_SRCS})
-
+ADD_LIBRARY(vtkoggtheora-gd ${VTKOGGTHEORA_SRCS})
+SET_TARGET_PROPERTIES(vtkoggtheora-gd PROPERTIES COMPILE_FLAGS ${CMAKE_CXX_FLAGS_DEBUG})
 # Apply user-defined properties to the library target.
 IF(VTK_LIBRARY_PROPERTIES)
   SET_TARGET_PROPERTIES(vtkoggtheora PROPERTIES ${VTK_LIBRARY_PROPERTIES}
+  SET_TARGET_PROPERTIES(vtkoggtheora-gd PROPERTIES ${VTK_LIBRARY_PROPERTIES}
     )
 ENDIF(VTK_LIBRARY_PROPERTIES)
 
@@ -261,6 +263,10 @@
     RUNTIME DESTINATION ${VTK_INSTALL_BIN_DIR_CM24} COMPONENT RuntimeLibraries
     LIBRARY DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT RuntimeLibraries
     ARCHIVE DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT Development)
+  INSTALL(TARGETS vtkoggtheora-gd
+    RUNTIME DESTINATION ${VTK_INSTALL_BIN_DIR_CM24} COMPONENT RuntimeLibraries
+    LIBRARY DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT RuntimeLibraries
+    ARCHIVE DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT Development)
 ENDIF(NOT VTK_INSTALL_NO_LIBRARIES)
 
 IF(NOT VTK_INSTALL_NO_DEVELOPMENT)
diff -ru VTK/Utilities/vtkpng/CMakeLists.txt VTK/Utilities/vtkpng/CMakeLists.txt
--- VTK/Utilities/vtkpng/CMakeLists.txt	2010-09-27 13:40:10.000000000 -0700
+++ VTK/Utilities/vtkpng/CMakeLists.txt	2012-02-23 20:47:52.000000000 -0800
@@ -26,17 +26,22 @@
                ${VTKPNG_BINARY_DIR}/pngDllConfig.h)
 
 ADD_LIBRARY(vtkpng ${PNG_SRCS})
+ADD_LIBRARY(vtkpng-gd ${PNG_SRCS})
+SET_TARGET_PROPERTIES(vtkpng-gd PROPERTIES COMPILE_FLAGS ${CMAKE_CXX_FLAGS_DEBUG})
 
 TARGET_LINK_LIBRARIES(vtkpng ${VTK_ZLIB_LIBRARIES})
+TARGET_LINK_LIBRARIES(vtkpng-gd ${VTK_ZLIB_LIBRARIES_DEBUG})
 
 # link against the math library
 IF(UNIX)
   TARGET_LINK_LIBRARIES(vtkpng -lm)
+  TARGET_LINK_LIBRARIES(vtkpng-gd -lm)
 ENDIF()
 
 # Apply user-defined properties to the library target.
 IF(VTK_LIBRARY_PROPERTIES)
   SET_TARGET_PROPERTIES(vtkpng PROPERTIES ${VTK_LIBRARY_PROPERTIES})
+  SET_TARGET_PROPERTIES(vtkpng-gd PROPERTIES ${VTK_LIBRARY_PROPERTIES})
 ENDIF(VTK_LIBRARY_PROPERTIES)
 
 IF(NOT VTK_INSTALL_NO_LIBRARIES)
@@ -44,6 +49,10 @@
     RUNTIME DESTINATION ${VTK_INSTALL_BIN_DIR_CM24} COMPONENT RuntimeLibraries
     LIBRARY DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT RuntimeLibraries
     ARCHIVE DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT Development)
+  INSTALL(TARGETS vtkpng-gd
+    RUNTIME DESTINATION ${VTK_INSTALL_BIN_DIR_CM24} COMPONENT RuntimeLibraries
+    LIBRARY DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT RuntimeLibraries
+    ARCHIVE DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT Development)
 ENDIF(NOT VTK_INSTALL_NO_LIBRARIES)
 IF(NOT VTK_INSTALL_NO_DEVELOPMENT)
   INSTALL(FILES
diff -ru VTK/Utilities/vtksqlite/CMakeLists.txt VTK/Utilities/vtksqlite/CMakeLists.txt
--- VTK/Utilities/vtksqlite/CMakeLists.txt	2010-09-27 13:40:10.000000000 -0700
+++ VTK/Utilities/vtksqlite/CMakeLists.txt	2012-02-23 20:47:52.000000000 -0800
@@ -28,6 +28,8 @@
 
 IF (WIN32)
   ADD_LIBRARY( vtksqlite STATIC ${SQLite_SRCS} )
+  ADD_LIBRARY( vtksqlite-gd STATIC ${SQLite_SRCS} )
+	SET_TARGET_PROPERTIES( vtksqlite-gd PROPERTIES COMPILE_FLAGS ${CMAKE_CXX_FLAGS_DEBUG} )
 ELSE (WIN32)
   INCLUDE(CheckIncludeFiles)
   CHECK_INCLUDE_FILES("pthread.h"        HAVE_PTHREAD_H)
@@ -35,13 +37,17 @@
     ADD_DEFINITIONS(-DHAVE_PTHREAD_H)
   ENDIF(HAVE_PTHREAD_H AND CMAKE_USE_PTHREADS)
   ADD_LIBRARY( vtksqlite ${SQLite_SRCS} )
+  ADD_LIBRARY( vtksqlite-gd ${SQLite_SRCS} )
+	SET_TARGET_PROPERTIES( vtksqlite-gd PROPERTIES COMPILE_FLAGS ${CMAKE_CXX_FLAGS_DEBUG} )
   IF (UNIX AND HAVE_PTHREAD_H AND CMAKE_USE_PTHREADS)
     TARGET_LINK_LIBRARIES(vtksqlite ${CMAKE_THREAD_LIBS})
+		TARGET_LINK_LIBRARIES(vtksqlite-gd ${CMAKE_THREAD_LIBS})
   ENDIF (UNIX AND HAVE_PTHREAD_H AND CMAKE_USE_PTHREADS)
 ENDIF (WIN32)
 
 IF(VTK_LIBRARY_PROPERTIES)
   SET_TARGET_PROPERTIES(vtksqlite PROPERTIES ${VTK_LIBRARY_PROPERTIES})
+  SET_TARGET_PROPERTIES(vtksqlite-gd PROPERTIES ${VTK_LIBRARY_PROPERTIES})
 ENDIF(VTK_LIBRARY_PROPERTIES)
 
 IF(NOT VTK_INSTALL_NO_LIBRARIES)
@@ -49,6 +55,10 @@
     RUNTIME DESTINATION ${VTK_INSTALL_BIN_DIR_CM24} COMPONENT RuntimeLibraries
     LIBRARY DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT RuntimeLibraries
     ARCHIVE DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT Development)
+  INSTALL(TARGETS vtksqlite-gd
+    RUNTIME DESTINATION ${VTK_INSTALL_BIN_DIR_CM24} COMPONENT RuntimeLibraries
+    LIBRARY DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT RuntimeLibraries
+    ARCHIVE DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT Development)
 ENDIF(NOT VTK_INSTALL_NO_LIBRARIES)
 
 IF(NOT VTK_INSTALL_NO_DEVELOPMENT)
diff -ru VTK/Utilities/vtktiff/CMakeLists.txt VTK/Utilities/vtktiff/CMakeLists.txt
--- VTK/Utilities/vtktiff/CMakeLists.txt	2010-09-27 13:40:10.000000000 -0700
+++ VTK/Utilities/vtktiff/CMakeLists.txt	2012-02-23 20:47:52.000000000 -0800
@@ -119,15 +119,20 @@
 
 ADD_LIBRARY(vtktiff ${common_SRCS})
 TARGET_LINK_LIBRARIES( vtktiff ${VTK_ZLIB_LIBRARIES} ${VTK_JPEG_LIBRARIES} )
+ADD_LIBRARY(vtktiff-gd ${common_SRCS})
+SET_TARGET_PROPERTIES( vtktiff-gd PROPERTIES COMPILE_FLAGS ${CMAKE_CXX_FLAGS_DEBUG})
+TARGET_LINK_LIBRARIES( vtktiff-gd ${VTK_ZLIB_LIBRARIES_DEBUG} ${VTK_JPEG_LIBRARIES_DEBUG} )
 
 # link against the math library
 IF(UNIX)
   TARGET_LINK_LIBRARIES(vtktiff -lm)
+  TARGET_LINK_LIBRARIES(vtktiff-gd -lm)
 ENDIF()
 
 # Apply user-defined properties to the library target.
 IF(VTK_LIBRARY_PROPERTIES)
   SET_TARGET_PROPERTIES(vtktiff PROPERTIES ${VTK_LIBRARY_PROPERTIES})
+  SET_TARGET_PROPERTIES(vtktiff-gd PROPERTIES ${VTK_LIBRARY_PROPERTIES})
 ENDIF(VTK_LIBRARY_PROPERTIES)
 
 IF(NOT VTK_INSTALL_NO_LIBRARIES)
@@ -135,6 +140,10 @@
     RUNTIME DESTINATION ${VTK_INSTALL_BIN_DIR_CM24} COMPONENT RuntimeLibraries
     LIBRARY DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT RuntimeLibraries
     ARCHIVE DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT Development)
+  INSTALL(TARGETS vtktiff-gd
+    RUNTIME DESTINATION ${VTK_INSTALL_BIN_DIR_CM24} COMPONENT RuntimeLibraries
+    LIBRARY DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT RuntimeLibraries
+    ARCHIVE DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT Development)
 ENDIF(NOT VTK_INSTALL_NO_LIBRARIES)
 IF(NOT VTK_INSTALL_NO_DEVELOPMENT)
   INSTALL(FILES
diff -ru VTK/Utilities/vtkzlib/CMakeLists.txt VTK/Utilities/vtkzlib/CMakeLists.txt
--- VTK/Utilities/vtkzlib/CMakeLists.txt	2010-09-27 13:40:10.000000000 -0700
+++ VTK/Utilities/vtkzlib/CMakeLists.txt	2012-02-23 20:47:52.000000000 -0800
@@ -30,10 +30,13 @@
 
 
 ADD_LIBRARY(vtkzlib ${ZLIB_SRCS})
+ADD_LIBRARY(vtkzlib-gd ${ZLIB_SRCS})
+set_target_properties(vtkzlib-gd PROPERTIES COMPILE_FLAGS ${CMAKE_CXX_FLAGS_DEBUG})
 
 # Apply user-defined properties to the library target.
 IF(VTK_LIBRARY_PROPERTIES)
   SET_TARGET_PROPERTIES(vtkzlib PROPERTIES ${VTK_LIBRARY_PROPERTIES})
+  SET_TARGET_PROPERTIES(vtkzlib-gd PROPERTIES ${VTK_LIBRARY_PROPERTIES})
 ENDIF(VTK_LIBRARY_PROPERTIES)
 
 IF(NOT VTK_INSTALL_NO_LIBRARIES)
@@ -41,6 +44,10 @@
     RUNTIME DESTINATION ${VTK_INSTALL_BIN_DIR_CM24} COMPONENT RuntimeLibraries
     LIBRARY DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT RuntimeLibraries
     ARCHIVE DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT Development)
+  INSTALL(TARGETS vtkzlib-gd
+    RUNTIME DESTINATION ${VTK_INSTALL_BIN_DIR_CM24} COMPONENT RuntimeLibraries
+    LIBRARY DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT RuntimeLibraries
+    ARCHIVE DESTINATION ${VTK_INSTALL_LIB_DIR_CM24} COMPONENT Development)
 ENDIF(NOT VTK_INSTALL_NO_LIBRARIES)
 
 IF(NOT VTK_INSTALL_NO_DEVELOPMENT)
diff -ru VTK/Views/CMakeLists.txt VTK/Views/CMakeLists.txt
--- VTK/Views/CMakeLists.txt	2010-09-27 13:40:10.000000000 -0700
+++ VTK/Views/CMakeLists.txt	2012-02-23 20:47:52.000000000 -0800
@@ -4,6 +4,7 @@
 SET(KIT_PYTHON_LIBS vtkInfovisPythonD)
 SET(KIT_JAVA_LIBS vtkInfovisJava)
 SET(KIT_INTERFACE_LIBRARIES vtkInfovis)
+SET(KIT_INTERFACE_LIBRARIES_DEBUG vtkInfovis-gd)
 
 SET( Kit_SRCS
 vtkConvertSelectionDomain.cxx
diff -ru VTK/VolumeRendering/CMakeLists.txt VTK/VolumeRendering/CMakeLists.txt
--- VTK/VolumeRendering/CMakeLists.txt	2010-09-27 13:40:11.000000000 -0700
+++ VTK/VolumeRendering/CMakeLists.txt	2012-02-23 20:47:52.000000000 -0800
@@ -5,7 +5,9 @@
 SET(KIT_PYTHON_LIBS vtkRenderingPythonD vtkIOPythonD)
 SET(KIT_JAVA_LIBS vtkRenderingJava vtkIOJava)
 SET(KIT_INTERFACE_LIBRARIES vtkRendering vtkIO)
+SET(KIT_INTERFACE_LIBRARIES_DEBUG vtkRendering-gd vtkIO-gd)
 SET(KIT_LIBS ${OPENGL_gl_LIBRARY})
+SET(KIT_LIBS_DEBUG ${OPENGL_gl_LIBRARY})
 SET(Kit_EXTRA_SRCS)
 SET(Kit_EXTRA_CMDS)
 SET(Kit_TCL_EXTRA_SRCS)
@@ -192,14 +194,17 @@
                  vtkMesaVolumeTextureMapper2D.cxx
     )
     SET (KIT_LIBS ${KIT_LIBS} ${MANGLED_MESA_LIBRARY})
+		SET (KIT_LIBS_DEBUG ${KIT_LIBS_DEBUG} ${MANGLED_MESA_LIBRARY})
     IF(MANGLED_OSMESA_LIBRARY)
       SET( KIT_LIBS ${KIT_LIBS} ${MANGLED_OSMESA_LIBRARY} )
+			SET( KIT_LIBS_DEBUG ${KIT_LIBS_DEBUG} ${MANGLED_OSMESA_LIBRARY} )
     ENDIF(MANGLED_OSMESA_LIBRARY)
   ENDIF (MANGLED_MESA_LIBRARY)
 ELSE (VTK_USE_MANGLED_MESA)
   IF(VTK_OPENGL_HAS_OSMESA)
     IF (OSMESA_LIBRARY)
       SET(KIT_LIBS ${KIT_LIBS} ${OSMESA_LIBRARY})
+			SET(KIT_LIBS_DEBUG ${KIT_LIBS_DEBUG} ${OSMESA_LIBRARY})
     ENDIF (OSMESA_LIBRARY)
   ENDIF(VTK_OPENGL_HAS_OSMESA)
 ENDIF(VTK_USE_MANGLED_MESA)
@@ -221,6 +226,7 @@
     )
   INCLUDE_DIRECTORIES (${VLI_INCLUDE_PATH_FOR_VP1000})
   SET (KIT_LIBS ${KIT_LIBS} ${VLI_LIBRARY_FOR_VP1000})
+	SET (KIT_LIBS_DEBUG ${KIT_LIBS_DEBUG} ${VLI_LIBRARY_FOR_VP1000})
 ENDIF (VTK_USE_VOLUMEPRO_1000)
 
 MARK_AS_ADVANCED (
diff -ru VTK/VTKCPack.cmake VTK/VTKCPack.cmake
--- VTK/VTKCPack.cmake	2010-09-27 13:40:10.000000000 -0700
+++ VTK/VTKCPack.cmake	2012-02-23 20:47:52.000000000 -0800
@@ -7,8 +7,15 @@
 
     # For now, only build the CPack installer if vtk(.exe) will be available for
     # installation:
-    #
-    IF(VTK_WRAP_TCL)
+    #	 
+    # define stuff for the PackageMaker packages
+      set(CPACK_OSX_PACKAGE_VERSION 10.5)
+      set(CPACK_PACKAGE_CONTACT "pcl-developers@pointclouds.org")
+      set(CPACK_SET_DESTDIR ON)
+      set(CPACK_PACKAGING_INSTALL_PREFIX /usr/local)
+      set(CPACK_GENERATOR PackageMaker)
+      set(CPACK_MONOLITHIC_INSTALL 1)
+      set(CPACK_PACKAGE_CONTACT "pointclouds.org")
       SET(CPACK_PACKAGE_DESCRIPTION_SUMMARY "VTK - The Visualization Toolkit")
       SET(CPACK_PACKAGE_VENDOR "Kitware, Inc.")
       SET(CPACK_PACKAGE_DESCRIPTION_FILE "${CMAKE_CURRENT_SOURCE_DIR}/Copyright.txt")
@@ -33,7 +40,6 @@
       ENDIF(WIN32)
 
       INCLUDE(CPack)
-    ENDIF(VTK_WRAP_TCL)
 
   ENDIF("${VTK_BINARY_DIR}" STREQUAL "${CMAKE_BINARY_DIR}")
 ENDIF(EXISTS "${CMAKE_ROOT}/Modules/CPack.cmake")
diff -ru VTK/vtkMacros.cmake VTK/vtkMacros.cmake
--- VTK/vtkMacros.cmake	2010-09-27 13:40:11.000000000 -0700
+++ VTK/vtkMacros.cmake	2012-02-23 20:47:52.000000000 -0800
@@ -19,6 +19,8 @@
                         HEADER_LOCATION ${VTK_BINARY_DIR})
   ENDIF (COMMAND VTK_MAKE_INSTANTIATOR2)
   ADD_LIBRARY(vtk${KIT}${VTK_VERSION} ${${KIT}_SRCS} ${Instantiator_SRCS})
+	ADD_LIBRARY(vtk${KIT}${VTK_VERSION}-gd ${${KIT}_SRCS} ${Instantiator_SRCS})
+	SET_TARGET_PROPERTIES(vtk${KIT}${VTK_VERSION}-gd PROPERTIES COMPILE_FLAGS ${CMAKE_CXX_FLAGS_DEBUG})
 ENDMACRO(VTK_USE_INSTANTIATOR KIT CAPS_KIT)
 
 #
diff -ru VTK/Widgets/CMakeLists.txt VTK/Widgets/CMakeLists.txt
--- VTK/Widgets/CMakeLists.txt	2010-09-27 13:40:11.000000000 -0700
+++ VTK/Widgets/CMakeLists.txt	2012-02-23 20:47:52.000000000 -0800
@@ -5,7 +5,9 @@
 SET(KIT_PYTHON_LIBS vtkRenderingPythonD vtkHybridPythonD)
 SET(KIT_JAVA_LIBS vtkRenderingJava vtkHybridJava)
 SET(KIT_INTERFACE_LIBRARIES vtkRendering vtkHybrid)
+SET(KIT_INTERFACE_LIBRARIES_DEBUG vtkRendering-gd vtkHybrid-gd)
 SET(KIT_LIBS ${OPENGL_gl_LIBRARY})
+SET(KIT_LIBS_DEBUG ${OPENGL_gl_LIBRARY})
 
 SET ( Kit_SRCS
 vtk3DWidget.cxx
